<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Partial Least Squares Regression for Cox Models and Related Techniques • plsRcox</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Partial Least Squares Regression for Cox Models and Related Techniques">
<meta property="og:description" content="Provides Partial least squares Regression and various regular, sparse or kernel, techniques for fitting Cox models in high dimensional settings &lt;doi:10.1093/bioinformatics/btu660&gt;, Bastien, P., Bertrand, F., Meyer N., Maumy-Bertrand, M. (2015), Deviance residuals-based sparse PLS and sparse kernel PLS regression for censored data, Bioinformatics, 31(3):397-404. Cross validation criteria were studied in &lt;arXiv:1810.02962&gt;, Bertrand, F., Bastien, Ph. and Maumy-Bertrand, M. (2018), Cross validating extensions of kernel, sparse or regular partial least squares regression models to censored data.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">plsRcox</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.7.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fbertran/plsRcox">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    

    
    

<!-- README.md is generated from README.Rmd. Please edit that file -->
<div id="plsrcox" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#plsrcox" class="anchor"></a>plsRcox</h1></div>
<p>The goal of plsRcox is provide Cox models in a high dimensional setting in R.</p>
<p>plsRcox implements partial least squares Regression and various regular, sparse or kernel, techniques for fitting Cox models in high dimensional settings <a href="https://doi.org/10.1093/bioinformatics/btu660" class="uri">https://doi.org/10.1093/bioinformatics/btu660</a>, Bastien, P., Bertrand, F., Meyer N., Maumy-Bertrand, M. (2015), Deviance residuals-based sparse PLS and sparse kernel PLS regression for censored data, Bioinformatics, 31(3):397-404. Cross validation criteria were studied in <a href="https://arxiv.org/abs/1810.01005">&lt;arXiv:1810.02962&gt;</a>, Bertrand, F., Bastien, Ph. and Maumy-Bertrand, M. (2018), Cross validating extensions of kernel, sparse or regular partial least squares regression models to censored data.</p>
<p>The package was presented at the <a href="http://user2014.r-project.org/">User2014!</a> conference. Frédéric Bertrand, Jérémy Magnanensi, Nicolas Meyer and Myriam Bertrand (2014). “plsRglm, PLS generalized linear models for R”, <em>book of abstracts</em>, User2014!, Los Angeles, page 152, <a href="http://user2014.r-project.org/abstracts/posters/172_Bertrand.pdf" class="uri">http://user2014.r-project.org/abstracts/posters/172_Bertrand.pdf</a>.</p>
<p>The plsRcox package contains an original allelotyping dataset from “Allelotyping identification of genomic alterations in rectal chromosomally unstable tumors without preoperative treatment”, Benoît Romain, Agnès Neuville, Nicolas Meyer, Cécile Brigand, Serge Rohr, Anne Schneider, Marie-Pierre Gaub and Dominique Guenot (2010), <em>BMC Cancer</em>, <strong>10</strong>:561, <a href="https://doi.org/10.1186/1471-2407-10-561" class="uri">https://doi.org/10.1186/1471-2407-10-561</a>.</p>
<p>Support for parallel computation and GPU is being developped.</p>
<p>The package provides several modelling techniques related to penalized Cox models or extensions of partial least squares to Cox models. The first two were new algorithms.</p>
<ul>
<li><p><strong>coxsplsDR</strong> and <strong>cv.coxsplsDR</strong> (Philippe Bastien, Frederic Bertrand, Nicolas Meyer, and Myriam Maumy-Bertrand (2015), “Deviance residuals-based sparse PLS and sparse kernel PLS regression for censored data”, <em>Bioinformatics</em>, <strong>31</strong>(3):397-404, <a href="https://doi.org/10.1093/bioinformatics/btu660" class="uri">https://doi.org/10.1093/bioinformatics/btu660</a>),</p></li>
<li><p><strong>coxDKsplsDR</strong> and <strong>cv.coxDKsplsDR</strong> (Philippe Bastien, Frederic Bertrand, Nicolas Meyer, and Myriam Maumy-Bertrand (2015), “Deviance residuals-based sparse PLS and sparse kernel PLS regression for censored data”, <em>Bioinformatics</em>, <strong>31</strong>(3):397-404, <a href="https://doi.org/10.1093/bioinformatics/btu660" class="uri">https://doi.org/10.1093/bioinformatics/btu660</a>),</p></li>
<li><p><strong>coxDKplsDR</strong> and <strong>cv.coxDKplsDR</strong> (Philippe Bastien (2008), “Deviance residuals based PLS regression for censored data in high dimensional setting”, <em>Chemometrics and Intelligent Laboratory Systems</em>, <strong>91</strong>:78–86, <a href="https://doi.org/10.1016/j.chemolab.2007.09.009" class="uri">https://doi.org/10.1016/j.chemolab.2007.09.009</a>),</p></li>
<li><p><strong>coxpls</strong> and <strong>cv.coxpls</strong> (Nguyen, D.V., Rocke, D.M. (2002), “Partial least squares proportional hazard regression for application to DNA microarray survival data”, <em>Bioinformatics</em>, <strong>18</strong>(12):1625–1632),</p></li>
<li><p><strong>coxplsDR</strong> and <strong>cv.coxplsDR</strong> (Philippe Bastien (2008), “Deviance residuals based PLS regression for censored data in high dimensional setting”, <em>Chemometrics and Intelligent Laboratory Systems</em>, <strong>91</strong>:78–86, <a href="https://doi.org/10.1016/j.chemolab.2007.09.009" class="uri">https://doi.org/10.1016/j.chemolab.2007.09.009</a>),</p></li>
<li><p><strong>DKplsRcox</strong>,</p></li>
<li><p><strong>larsDR</strong> and <strong>cv.larsDR</strong> (Segal, M.R. (2006), “Microarray Gene Expression Data with Linked Survival Phenotypes: Diffuse large-B- Cell Lymphoma Revisited”, <em>Biostatistics</em>, <strong>7</strong>:268-285, <a href="https://doi.org/10.1093/biostatistics/kxj006" class="uri">https://doi.org/10.1093/biostatistics/kxj006</a>),</p></li>
<li><p><strong>plsRcox</strong> and <strong>cv.plsRcox</strong> (Philippe Bastien, Vincenzo Esposito Vinzi, and Michel Tenenhaus (2005), “PLS generalised linear regression”, <em>Computational Statistics &amp; Data Analysis</em>, <strong>48</strong>(1):17–46, <a href="https://doi.org/10.1016/j.csda.2004.02.005" class="uri">https://doi.org/10.1016/j.csda.2004.02.005</a>),</p></li>
<li><p><strong>autoplsRcox</strong> and <strong>cv.autoplsRcox</strong> (Philippe Bastien, Vincenzo Esposito Vinzi, and Michel Tenenhaus (2005), “PLS generalised linear regression”, <em>Computational Statistics &amp; Data Analysis</em>, <strong>48</strong>(1):17–46, <a href="https://doi.org/10.1016/j.csda.2004.02.005" class="uri">https://doi.org/10.1016/j.csda.2004.02.005</a>),</p></li>
</ul>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>You can install the released version of plsRcox from <a href="https://CRAN.R-project.org">CRAN</a> with:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="st">"plsRcox"</span>)</code></pre></div>
<p>You can install the development version of plsRcox from <a href="https://github.com">github</a> with:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"fbertran/plsRcox"</span>)</code></pre></div>
</div>
<div id="example" class="section level2">
<h2 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h2>
<div id="the-original-allelotyping-dataset" class="section level3">
<h3 class="hasAnchor">
<a href="#the-original-allelotyping-dataset" class="anchor"></a>The original allelotyping dataset</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(plsRcox)
<span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(micro.censure)
Y_train_micro &lt;-<span class="st"> </span>micro.censure<span class="op">$</span>survyear[<span class="dv">1</span><span class="op">:</span><span class="dv">80</span>]
C_train_micro &lt;-<span class="st"> </span>micro.censure<span class="op">$</span>DC[<span class="dv">1</span><span class="op">:</span><span class="dv">80</span>]
Y_test_micro &lt;-<span class="st"> </span>micro.censure<span class="op">$</span>survyear[<span class="dv">81</span><span class="op">:</span><span class="dv">117</span>]
C_test_micro &lt;-<span class="st"> </span>micro.censure<span class="op">$</span>DC[<span class="dv">81</span><span class="op">:</span><span class="dv">117</span>]

<span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(Xmicro.censure_compl_imp)
X_train_micro &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/apply">apply</a></span>((<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">as.matrix</a></span>(Xmicro.censure_compl_imp)),<span class="dt">FUN=</span><span class="st">"as.numeric"</span>,<span class="dt">MARGIN=</span><span class="dv">2</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">80</span>,]
X_train_micro_df &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(X_train_micro)</code></pre></div>
<p>Compute deviance residuals with some options.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/DR_coxph.html">DR_coxph</a></span>(Y_train_micro,C_train_micro,<span class="dt">plot=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><img src="reference/figures/README-devianceresiduals-1.png" width="100%"></p>
<pre><code>#&gt;           1           2           3           4           5           6 
#&gt; -1.48432960 -0.54695398 -0.23145502 -0.34003013 -0.97633722 -0.38667660 
#&gt;           7           8           9          10          11          12 
#&gt; -0.38667660  1.57418914 -0.54695398 -0.15811388  2.10405254 -0.23145502 
#&gt;          13          14          15          16          17          18 
#&gt; -0.38667660 -1.09692040 -0.15811388 -0.15811388 -0.54695398 -0.38667660 
#&gt;          19          20          21          22          23          24 
#&gt;  0.65978609 -1.09692040 -0.43627414 -0.28961087 -0.38667660 -0.97633722 
#&gt;          25          26          27          28          29          30 
#&gt; -1.09692040 -0.15811388 -0.43627414 -0.43627414 -0.38667660 -0.23145502 
#&gt;          31          32          33          34          35          36 
#&gt;  2.30072697 -0.49023986 -0.54695398 -0.73444882  1.31082939 -0.97633722 
#&gt;          37          38          39          40          41          42 
#&gt;  1.70134282 -0.54695398 -0.15811388  1.07714870 -0.15811388 -0.49023986 
#&gt;          43          44          45          46          47          48 
#&gt; -0.34003013 -0.97633722 -0.15811388 -0.91410465 -1.09692040 -0.43627414 
#&gt;          49          50          51          52          53          54 
#&gt; -0.38667660 -0.09836581 -0.79392956  0.46851068 -0.34003013  1.95366297 
#&gt;          55          56          57          58          59          60 
#&gt;  2.60558118 -0.54695398 -1.09692040 -0.15811388 -0.49023986 -0.97633722 
#&gt;          61          62          63          64          65          66 
#&gt; -0.28961087  1.44879795  1.82660327 -0.38667660  0.96936094 -0.15811388 
#&gt;          67          68          69          70          71          72 
#&gt; -0.43627414 -0.49023986  1.18850436 -0.97633722 -0.97633722  0.86322194 
#&gt;          73          74          75          76          77          78 
#&gt; -0.43627414 -0.49023986 -0.38667660  0.76231394 -0.97633722 -0.43627414 
#&gt;          79          80 
#&gt; -0.54695398 -0.43627414</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/DR_coxph.html">DR_coxph</a></span>(Y_train_micro,C_train_micro,<span class="dt">scaleY=</span><span class="ot">FALSE</span>,<span class="dt">plot=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><img src="reference/figures/README-devianceresiduals2-1.png" width="100%"></p>
<pre><code>#&gt;           1           2           3           4           5           6 
#&gt; -1.48432960 -0.54695398 -0.23145502 -0.34003013 -0.97633722 -0.38667660 
#&gt;           7           8           9          10          11          12 
#&gt; -0.38667660  1.57418914 -0.54695398 -0.15811388  2.10405254 -0.23145502 
#&gt;          13          14          15          16          17          18 
#&gt; -0.38667660 -1.09692040 -0.15811388 -0.15811388 -0.54695398 -0.38667660 
#&gt;          19          20          21          22          23          24 
#&gt;  0.65978609 -1.09692040 -0.43627414 -0.28961087 -0.38667660 -0.97633722 
#&gt;          25          26          27          28          29          30 
#&gt; -1.09692040 -0.15811388 -0.43627414 -0.43627414 -0.38667660 -0.23145502 
#&gt;          31          32          33          34          35          36 
#&gt;  2.30072697 -0.49023986 -0.54695398 -0.73444882  1.31082939 -0.97633722 
#&gt;          37          38          39          40          41          42 
#&gt;  1.70134282 -0.54695398 -0.15811388  1.07714870 -0.15811388 -0.49023986 
#&gt;          43          44          45          46          47          48 
#&gt; -0.34003013 -0.97633722 -0.15811388 -0.91410465 -1.09692040 -0.43627414 
#&gt;          49          50          51          52          53          54 
#&gt; -0.38667660 -0.09836581 -0.79392956  0.46851068 -0.34003013  1.95366297 
#&gt;          55          56          57          58          59          60 
#&gt;  2.60558118 -0.54695398 -1.09692040 -0.15811388 -0.49023986 -0.97633722 
#&gt;          61          62          63          64          65          66 
#&gt; -0.28961087  1.44879795  1.82660327 -0.38667660  0.96936094 -0.15811388 
#&gt;          67          68          69          70          71          72 
#&gt; -0.43627414 -0.49023986  1.18850436 -0.97633722 -0.97633722  0.86322194 
#&gt;          73          74          75          76          77          78 
#&gt; -0.43627414 -0.49023986 -0.38667660  0.76231394 -0.97633722 -0.43627414 
#&gt;          79          80 
#&gt; -0.54695398 -0.43627414</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/DR_coxph.html">DR_coxph</a></span>(Y_train_micro,C_train_micro,<span class="dt">scaleY=</span><span class="ot">TRUE</span>,<span class="dt">plot=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><img src="reference/figures/README-devianceresiduals3-1.png" width="100%"></p>
<pre><code>#&gt;           1           2           3           4           5           6 
#&gt; -1.48432960 -0.54695398 -0.23145502 -0.34003013 -0.97633722 -0.38667660 
#&gt;           7           8           9          10          11          12 
#&gt; -0.38667660  1.57418914 -0.54695398 -0.15811388  2.10405254 -0.23145502 
#&gt;          13          14          15          16          17          18 
#&gt; -0.38667660 -1.09692040 -0.15811388 -0.15811388 -0.54695398 -0.38667660 
#&gt;          19          20          21          22          23          24 
#&gt;  0.65978609 -1.09692040 -0.43627414 -0.28961087 -0.38667660 -0.97633722 
#&gt;          25          26          27          28          29          30 
#&gt; -1.09692040 -0.15811388 -0.43627414 -0.43627414 -0.38667660 -0.23145502 
#&gt;          31          32          33          34          35          36 
#&gt;  2.30072697 -0.49023986 -0.54695398 -0.73444882  1.31082939 -0.97633722 
#&gt;          37          38          39          40          41          42 
#&gt;  1.70134282 -0.54695398 -0.15811388  1.07714870 -0.15811388 -0.49023986 
#&gt;          43          44          45          46          47          48 
#&gt; -0.34003013 -0.97633722 -0.15811388 -0.91410465 -1.09692040 -0.43627414 
#&gt;          49          50          51          52          53          54 
#&gt; -0.38667660 -0.09836581 -0.79392956  0.46851068 -0.34003013  1.95366297 
#&gt;          55          56          57          58          59          60 
#&gt;  2.60558118 -0.54695398 -1.09692040 -0.15811388 -0.49023986 -0.97633722 
#&gt;          61          62          63          64          65          66 
#&gt; -0.28961087  1.44879795  1.82660327 -0.38667660  0.96936094 -0.15811388 
#&gt;          67          68          69          70          71          72 
#&gt; -0.43627414 -0.49023986  1.18850436 -0.97633722 -0.97633722  0.86322194 
#&gt;          73          74          75          76          77          78 
#&gt; -0.43627414 -0.49023986 -0.38667660  0.76231394 -0.97633722 -0.43627414 
#&gt;          79          80 
#&gt; -0.54695398 -0.43627414</code></pre>
</div>
<div id="coxsplsdr" class="section level3">
<h3 class="hasAnchor">
<a href="#coxsplsdr" class="anchor"></a>coxsplsDR</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(<span class="dt">cox_splsDR_fit=</span><span class="kw"><a href="reference/coxsplsDR.html">coxsplsDR</a></span>(X_train_micro,Y_train_micro,C_train_micro,<span class="dt">ncomp=</span><span class="dv">6</span>,<span class="dt">eta=</span>.<span class="dv">5</span>))
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; coxph(formula = YCsurv ~ ., data = tt_splsDR)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;         coef exp(coef) se(coef)     z        p</span>
<span class="co">#&gt; dim.1 0.8093    2.2462   0.2029 3.989 6.63e-05</span>
<span class="co">#&gt; dim.2 0.9295    2.5333   0.2939 3.163  0.00156</span>
<span class="co">#&gt; dim.3 0.9968    2.7096   0.4190 2.379  0.01736</span>
<span class="co">#&gt; dim.4 0.9705    2.6391   0.3793 2.558  0.01052</span>
<span class="co">#&gt; dim.5 0.2162    1.2413   0.2811 0.769  0.44192</span>
<span class="co">#&gt; dim.6 0.4380    1.5496   0.3608 1.214  0.22473</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Likelihood ratio test=55.06  on 6 df, p=4.51e-10</span>
<span class="co">#&gt; n= 80, number of events= 17</span>

(<span class="dt">cox_splsDR_fit2=</span><span class="kw"><a href="reference/coxsplsDR.html">coxsplsDR</a></span>(<span class="op">~</span>X_train_micro,Y_train_micro,C_train_micro,<span class="dt">ncomp=</span><span class="dv">6</span>,<span class="dt">eta=</span>.<span class="dv">5</span>,<span class="dt">trace=</span><span class="ot">TRUE</span>))
<span class="co">#&gt; The variables that join the set of selected variables at each step:</span>
<span class="co">#&gt; - 1th step (K=1):</span>
<span class="co">#&gt; X_train_microD20S107 X_train_microD5S346 X_train_microD1S225 X_train_microD3S1282 X_train_microD15S127 X_train_microD1S207 X_train_microD2S138 X_train_microD10S191 X_train_microD14S65 X_train_microD4S414</span>
<span class="co">#&gt; X_train_microD16S408 X_train_microT X_train_microN X_train_microSTADE</span>
<span class="co">#&gt; - 2th step (K=2):</span>
<span class="co">#&gt; X_train_microD22S928 X_train_microD16S422 X_train_microD3S1283 X_train_microAgediag X_train_microM</span>
<span class="co">#&gt; - 3th step (K=3):</span>
<span class="co">#&gt; X_train_microD1S305 X_train_microD8S283 X_train_microD10S192 X_train_microsexe X_train_microSiege</span>
<span class="co">#&gt; - 4th step (K=4):</span>
<span class="co">#&gt; X_train_microD17S794 X_train_microD13S173 X_train_microTP53 X_train_microD6S264 X_train_microD2S159 X_train_microD6S275</span>
<span class="co">#&gt; - 5th step (K=5):</span>
<span class="co">#&gt; X_train_microD18S61 X_train_microD9S171 X_train_microD8S264 X_train_microD18S53 X_train_microD4S394 X_train_microD11S916</span>
<span class="co">#&gt; - 6th step (K=6):</span>
<span class="co">#&gt; X_train_microD17S790</span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; coxph(formula = YCsurv ~ ., data = tt_splsDR)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;         coef exp(coef) se(coef)     z        p</span>
<span class="co">#&gt; dim.1 0.8093    2.2462   0.2029 3.989 6.63e-05</span>
<span class="co">#&gt; dim.2 0.9295    2.5333   0.2939 3.163  0.00156</span>
<span class="co">#&gt; dim.3 0.9968    2.7096   0.4190 2.379  0.01736</span>
<span class="co">#&gt; dim.4 0.9705    2.6391   0.3793 2.558  0.01052</span>
<span class="co">#&gt; dim.5 0.2162    1.2413   0.2811 0.769  0.44192</span>
<span class="co">#&gt; dim.6 0.4380    1.5496   0.3608 1.214  0.22473</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Likelihood ratio test=55.06  on 6 df, p=4.51e-10</span>
<span class="co">#&gt; n= 80, number of events= 17</span>

(<span class="dt">cox_splsDR_fit3=</span><span class="kw"><a href="reference/coxsplsDR.html">coxsplsDR</a></span>(<span class="op">~</span>.,Y_train_micro,C_train_micro,<span class="dt">ncomp=</span><span class="dv">6</span>,
<span class="dt">dataXplan=</span>X_train_micro_df,<span class="dt">eta=</span>.<span class="dv">5</span>))
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; coxph(formula = YCsurv ~ ., data = tt_splsDR)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;         coef exp(coef) se(coef)     z        p</span>
<span class="co">#&gt; dim.1 0.8093    2.2462   0.2029 3.989 6.63e-05</span>
<span class="co">#&gt; dim.2 0.9295    2.5333   0.2939 3.163  0.00156</span>
<span class="co">#&gt; dim.3 0.9968    2.7096   0.4190 2.379  0.01736</span>
<span class="co">#&gt; dim.4 0.9705    2.6391   0.3793 2.558  0.01052</span>
<span class="co">#&gt; dim.5 0.2162    1.2413   0.2811 0.769  0.44192</span>
<span class="co">#&gt; dim.6 0.4380    1.5496   0.3608 1.214  0.22473</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Likelihood ratio test=55.06  on 6 df, p=4.51e-10</span>
<span class="co">#&gt; n= 80, number of events= 17</span>

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rm">rm</a></span>(cox_splsDR_fit,cox_splsDR_fit2,cox_splsDR_fit3)</code></pre></div>
</div>
<div id="cv-coxsplsdr" class="section level3">
<h3 class="hasAnchor">
<a href="#cv-coxsplsdr" class="anchor"></a>cv.coxsplsDR</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">123456</span>)

(<span class="dt">cv.coxsplsDR.res=</span><span class="kw"><a href="reference/cv.coxsplsDR.html">cv.coxsplsDR</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">x=</span>X_train_micro,<span class="dt">time=</span>Y_train_micro, <span class="dt">status=</span>C_train_micro),<span class="dt">nt=</span><span class="dv">10</span>,<span class="dt">eta=</span>.<span class="dv">1</span>))
<span class="co">#&gt; CV Fold 1 </span>
<span class="co">#&gt; CV Fold 2 </span>
<span class="co">#&gt; CV Fold 3 </span>
<span class="co">#&gt; CV Fold 4 </span>
<span class="co">#&gt; CV Fold 5</span></code></pre></div>
<p><img src="reference/figures/README-cvcoxsplsDR-1.png" width="100%"></p>
<pre><code>#&gt; $nt
#&gt; [1] 10
#&gt; 
#&gt; $cv.error10
#&gt;  [1] 0.5000000 0.6773894 0.6493620 0.6641847 0.6240118 0.6442501 0.6368322
#&gt;  [8] 0.6286337 0.6116994 0.6125641 0.6488137
#&gt; 
#&gt; $cv.se10
#&gt;  [1] 0.00000000 0.02579921 0.04338024 0.03888348 0.04730986 0.04581284
#&gt;  [7] 0.05021484 0.04737293 0.05050230 0.04570564 0.04490152
#&gt; 
#&gt; $folds
#&gt; $folds$`1`
#&gt;  [1] 64 15 56 59 10 39 70  4 80 16 37 33 50 51 78 44
#&gt; 
#&gt; $folds$`2`
#&gt;  [1] 60 40 41 57  5 61 43 42 29 20 32 68  1 36 55 17
#&gt; 
#&gt; $folds$`3`
#&gt;  [1] 31  8 62 13 73 47 71 46 35 69 75 26 74 65 58 63
#&gt; 
#&gt; $folds$`4`
#&gt;  [1] 27 72 79 21 67 45  6 30 11 22 24 53 23  2 49  7
#&gt; 
#&gt; $folds$`5`
#&gt;  [1] 28 12 66 48 77  9 38 25 19 54 34 52  3 14 76 18
#&gt; 
#&gt; 
#&gt; $lambda.min10
#&gt; [1] 1
#&gt; 
#&gt; $lambda.1se10
#&gt; [1] 0
#&gt; 
#&gt; $nzb
#&gt;  [1]  0 32 37 40 40 40 40 40 40 40 40</code></pre>
</div>
<div id="coxdksplsdr" class="section level3">
<h3 class="hasAnchor">
<a href="#coxdksplsdr" class="anchor"></a>coxDKsplsDR</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(<span class="dt">cox_DKsplsDR_fit=</span><span class="kw"><a href="reference/coxDKsplsDR.html">coxDKsplsDR</a></span>(X_train_micro,Y_train_micro,C_train_micro,<span class="dt">ncomp=</span><span class="dv">6</span>, <span class="dt">validation=</span><span class="st">"CV"</span>,<span class="dt">eta=</span>.<span class="dv">5</span>))
<span class="co">#&gt; Kernel :  rbfdot </span>
<span class="co">#&gt; Estimated_sigma  0.01224473</span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; coxph(formula = YCsurv ~ ., data = tt_DKsplsDR)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;            coef exp(coef)  se(coef)     z       p</span>
<span class="co">#&gt; dim.1 3.637e+00 3.797e+01 1.246e+00 2.919 0.00351</span>
<span class="co">#&gt; dim.2 9.920e+00 2.033e+04 3.303e+00 3.003 0.00267</span>
<span class="co">#&gt; dim.3 6.502e+00 6.668e+02 2.579e+00 2.521 0.01170</span>
<span class="co">#&gt; dim.4 1.467e+01 2.342e+06 4.854e+00 3.021 0.00252</span>
<span class="co">#&gt; dim.5 6.109e+00 4.497e+02 2.759e+00 2.214 0.02682</span>
<span class="co">#&gt; dim.6 1.249e+01 2.666e+05 4.980e+00 2.509 0.01212</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Likelihood ratio test=69.58  on 6 df, p=4.977e-13</span>
<span class="co">#&gt; n= 80, number of events= 17</span>

(<span class="dt">cox_DKsplsDR_fit=</span><span class="kw"><a href="reference/coxDKsplsDR.html">coxDKsplsDR</a></span>(<span class="op">~</span>X_train_micro,Y_train_micro,C_train_micro,<span class="dt">ncomp=</span><span class="dv">6</span>, <span class="dt">validation=</span><span class="st">"CV"</span>,<span class="dt">eta=</span>.<span class="dv">5</span>))
<span class="co">#&gt; Kernel :  rbfdot </span>
<span class="co">#&gt; Estimated_sigma  0.01198748</span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; coxph(formula = YCsurv ~ ., data = tt_DKsplsDR)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;            coef exp(coef)  se(coef)     z        p</span>
<span class="co">#&gt; dim.1 2.649e+00 1.414e+01 1.206e+00 2.197 0.028012</span>
<span class="co">#&gt; dim.2 1.004e+01 2.304e+04 2.861e+00 3.511 0.000447</span>
<span class="co">#&gt; dim.3 6.887e+00 9.791e+02 2.476e+00 2.782 0.005404</span>
<span class="co">#&gt; dim.4 1.166e+01 1.160e+05 3.821e+00 3.052 0.002274</span>
<span class="co">#&gt; dim.5 8.334e+00 4.164e+03 4.227e+00 1.971 0.048674</span>
<span class="co">#&gt; dim.6 5.721e+00 3.051e+02 2.425e+00 2.359 0.018313</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Likelihood ratio test=64.42  on 6 df, p=5.67e-12</span>
<span class="co">#&gt; n= 80, number of events= 17</span>

(<span class="dt">cox_DKsplsDR_fit=</span><span class="kw"><a href="reference/coxDKsplsDR.html">coxDKsplsDR</a></span>(<span class="op">~</span>.,Y_train_micro,C_train_micro,<span class="dt">ncomp=</span><span class="dv">6</span>,
<span class="dt">validation=</span><span class="st">"CV"</span>,<span class="dt">dataXplan=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(X_train_micro),<span class="dt">eta=</span>.<span class="dv">5</span>))
<span class="co">#&gt; Kernel :  rbfdot </span>
<span class="co">#&gt; Estimated_sigma  0.0131134</span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; coxph(formula = YCsurv ~ ., data = tt_DKsplsDR)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;            coef exp(coef)  se(coef)     z       p</span>
<span class="co">#&gt; dim.1 4.409e+00 8.218e+01 1.449e+00 3.043 0.00234</span>
<span class="co">#&gt; dim.2 9.764e+00 1.739e+04 3.227e+00 3.025 0.00248</span>
<span class="co">#&gt; dim.3 6.298e+00 5.434e+02 2.382e+00 2.644 0.00819</span>
<span class="co">#&gt; dim.4 1.491e+01 2.994e+06 4.781e+00 3.119 0.00182</span>
<span class="co">#&gt; dim.5 4.592e+00 9.865e+01 2.267e+00 2.025 0.04284</span>
<span class="co">#&gt; dim.6 6.985e+00 1.080e+03 3.048e+00 2.291 0.02194</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Likelihood ratio test=69.14  on 6 df, p=6.149e-13</span>
<span class="co">#&gt; n= 80, number of events= 17</span>

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rm">rm</a></span>(cox_DKsplsDR_fit)</code></pre></div>
</div>
<div id="cv-coxsplsdr-1" class="section level3">
<h3 class="hasAnchor">
<a href="#cv-coxsplsdr-1" class="anchor"></a>cv.coxsplsDR</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">123456</span>)

(<span class="dt">cv.coxDKsplsDR.res=</span><span class="kw"><a href="reference/cv.coxDKsplsDR.html">cv.coxDKsplsDR</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">x=</span>X_train_micro,<span class="dt">time=</span>Y_train_micro, <span class="dt">status=</span>C_train_micro),<span class="dt">nt=</span><span class="dv">10</span>,<span class="dt">eta=</span>.<span class="dv">1</span>))
<span class="co">#&gt; Kernel :  rbfdot </span>
<span class="co">#&gt; Estimated_sigma  0.01186138 </span>
<span class="co">#&gt; CV Fold 1 </span>
<span class="co">#&gt; Kernel :  rbfdot </span>
<span class="co">#&gt; Estimated_sigma  0.0123956 </span>
<span class="co">#&gt; CV Fold 2 </span>
<span class="co">#&gt; Kernel :  rbfdot </span>
<span class="co">#&gt; Estimated_sigma  0.01263568 </span>
<span class="co">#&gt; CV Fold 3 </span>
<span class="co">#&gt; Kernel :  rbfdot </span>
<span class="co">#&gt; Estimated_sigma  0.01235948 </span>
<span class="co">#&gt; CV Fold 4 </span>
<span class="co">#&gt; Kernel :  rbfdot </span>
<span class="co">#&gt; Estimated_sigma  0.01247944 </span>
<span class="co">#&gt; CV Fold 5</span></code></pre></div>
<p><img src="reference/figures/README-cvcoxDKsplsDR-1.png" width="100%"></p>
<pre><code>#&gt; $nt
#&gt; [1] 10
#&gt; 
#&gt; $cv.error10
#&gt;  [1] 0.5000000 0.6452065 0.6390089 0.6596989 0.7176982 0.6872992 0.6713925
#&gt;  [8] 0.6911991 0.5932097 0.6453234 0.6199085
#&gt; 
#&gt; $cv.se10
#&gt;  [1] 0.00000000 0.01653815 0.04090982 0.03568890 0.04264094 0.04067596
#&gt;  [7] 0.04617710 0.03699745 0.03283319 0.03716281 0.03878102
#&gt; 
#&gt; $folds
#&gt; $folds$`1`
#&gt;  [1] 64 15 56 59 10 39 70  4 80 16 37 33 50 51 78 44
#&gt; 
#&gt; $folds$`2`
#&gt;  [1] 60 40 41 57  5 61 43 42 29 20 32 68  1 36 55 17
#&gt; 
#&gt; $folds$`3`
#&gt;  [1] 31  8 62 13 73 47 71 46 35 69 75 26 74 65 58 63
#&gt; 
#&gt; $folds$`4`
#&gt;  [1] 27 72 79 21 67 45  6 30 11 22 24 53 23  2 49  7
#&gt; 
#&gt; $folds$`5`
#&gt;  [1] 28 12 66 48 77  9 38 25 19 54 34 52  3 14 76 18
#&gt; 
#&gt; 
#&gt; $lambda.min10
#&gt; [1] 4
#&gt; 
#&gt; $lambda.1se10
#&gt; [1] 0
#&gt; 
#&gt; $nzb
#&gt;  [1]  0 53 62 64 64 64 64 64 64 64 64</code></pre>
</div>
</div>
<div id="plsrcox-1" class="section level2">
<h2 class="hasAnchor">
<a href="#plsrcox-1" class="anchor"></a>plsRcox</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/plsRcox.html">plsRcox</a></span>(X_train_micro,<span class="dt">time=</span>Y_train_micro,<span class="dt">event=</span>C_train_micro,<span class="dt">nt=</span><span class="dv">5</span>)
<span class="co">#&gt; ____************************************************____</span>
<span class="co">#&gt; ____Component____ 1 ____</span>
<span class="co">#&gt; ____Component____ 2 ____</span>
<span class="co">#&gt; ____Component____ 3 ____</span>
<span class="co">#&gt; ____Component____ 4 ____</span>
<span class="co">#&gt; ____Component____ 5 ____</span>
<span class="co">#&gt; ____Predicting X without NA neither in X nor in Y____</span>
<span class="co">#&gt; ****________________________________________________****</span>
<span class="co">#&gt; Number of required components:</span>
<span class="co">#&gt; [1] 5</span>
<span class="co">#&gt; Number of successfully computed components:</span>
<span class="co">#&gt; [1] 5</span>
<span class="co">#&gt; Coefficients:</span>
<span class="co">#&gt;                [,1]</span>
<span class="co">#&gt; D18S61   0.68964919</span>
<span class="co">#&gt; D17S794 -1.14362392</span>
<span class="co">#&gt; D13S173  1.37632457</span>
<span class="co">#&gt; D20S107  4.96128745</span>
<span class="co">#&gt; TP53     1.68453950</span>
<span class="co">#&gt; D9S171  -1.46691252</span>
<span class="co">#&gt; D8S264   0.66710776</span>
<span class="co">#&gt; D5S346  -4.61338196</span>
<span class="co">#&gt; D22S928 -1.82005524</span>
<span class="co">#&gt; D18S53   0.79853646</span>
<span class="co">#&gt; D1S225  -1.46234986</span>
<span class="co">#&gt; D3S1282 -1.67925042</span>
<span class="co">#&gt; D15S127  3.92225537</span>
<span class="co">#&gt; D1S305  -2.29680161</span>
<span class="co">#&gt; D1S207   2.02539691</span>
<span class="co">#&gt; D2S138  -3.48975878</span>
<span class="co">#&gt; D16S422 -2.92189625</span>
<span class="co">#&gt; D9S179  -0.59484679</span>
<span class="co">#&gt; D10S191 -1.30136747</span>
<span class="co">#&gt; D4S394   1.34265359</span>
<span class="co">#&gt; D1S197  -0.75014044</span>
<span class="co">#&gt; D6S264   1.32746604</span>
<span class="co">#&gt; D14S65  -3.20882866</span>
<span class="co">#&gt; D17S790  0.55427680</span>
<span class="co">#&gt; D5S430   3.40654627</span>
<span class="co">#&gt; D3S1283  2.12510239</span>
<span class="co">#&gt; D4S414   2.73619967</span>
<span class="co">#&gt; D8S283   0.71955323</span>
<span class="co">#&gt; D11S916  1.45026508</span>
<span class="co">#&gt; D2S159   0.90293134</span>
<span class="co">#&gt; D16S408 -0.59719901</span>
<span class="co">#&gt; D6S275  -1.02204186</span>
<span class="co">#&gt; D10S192  1.14220367</span>
<span class="co">#&gt; sexe     0.67314561</span>
<span class="co">#&gt; Agediag  0.04908478</span>
<span class="co">#&gt; Siege   -0.41985924</span>
<span class="co">#&gt; T        2.70581463</span>
<span class="co">#&gt; N        2.47039973</span>
<span class="co">#&gt; M       -4.53213922</span>
<span class="co">#&gt; STADE    0.48221697</span>
<span class="co">#&gt; Information criteria and Fit statistics:</span>
<span class="co">#&gt;                 AIC       BIC</span>
<span class="co">#&gt; Nb_Comp_0 112.87990 112.87990</span>
<span class="co">#&gt; Nb_Comp_1  85.11075  87.49278</span>
<span class="co">#&gt; Nb_Comp_2  75.49537  80.25942</span>
<span class="co">#&gt; Nb_Comp_3  68.45852  75.60460</span>
<span class="co">#&gt; Nb_Comp_4  63.09284  72.62094</span>
<span class="co">#&gt; Nb_Comp_5  55.30567  67.21581</span>

<span class="kw"><a href="reference/plsRcox.html">plsRcox</a></span>(<span class="op">~</span>X_train_micro,<span class="dt">time=</span>Y_train_micro,<span class="dt">event=</span>C_train_micro,<span class="dt">nt=</span><span class="dv">5</span>)
<span class="co">#&gt; ____************************************************____</span>
<span class="co">#&gt; ____Component____ 1 ____</span>
<span class="co">#&gt; ____Component____ 2 ____</span>
<span class="co">#&gt; ____Component____ 3 ____</span>
<span class="co">#&gt; ____Component____ 4 ____</span>
<span class="co">#&gt; ____Component____ 5 ____</span>
<span class="co">#&gt; ____Predicting X without NA neither in X nor in Y____</span>
<span class="co">#&gt; ****________________________________________________****</span>
<span class="co">#&gt; Number of required components:</span>
<span class="co">#&gt; [1] 5</span>
<span class="co">#&gt; Number of successfully computed components:</span>
<span class="co">#&gt; [1] 5</span>
<span class="co">#&gt; Coefficients:</span>
<span class="co">#&gt;                             [,1]</span>
<span class="co">#&gt; X_train_microD18S61   0.68964919</span>
<span class="co">#&gt; X_train_microD17S794 -1.14362392</span>
<span class="co">#&gt; X_train_microD13S173  1.37632457</span>
<span class="co">#&gt; X_train_microD20S107  4.96128745</span>
<span class="co">#&gt; X_train_microTP53     1.68453950</span>
<span class="co">#&gt; X_train_microD9S171  -1.46691252</span>
<span class="co">#&gt; X_train_microD8S264   0.66710776</span>
<span class="co">#&gt; X_train_microD5S346  -4.61338196</span>
<span class="co">#&gt; X_train_microD22S928 -1.82005524</span>
<span class="co">#&gt; X_train_microD18S53   0.79853646</span>
<span class="co">#&gt; X_train_microD1S225  -1.46234986</span>
<span class="co">#&gt; X_train_microD3S1282 -1.67925042</span>
<span class="co">#&gt; X_train_microD15S127  3.92225537</span>
<span class="co">#&gt; X_train_microD1S305  -2.29680161</span>
<span class="co">#&gt; X_train_microD1S207   2.02539691</span>
<span class="co">#&gt; X_train_microD2S138  -3.48975878</span>
<span class="co">#&gt; X_train_microD16S422 -2.92189625</span>
<span class="co">#&gt; X_train_microD9S179  -0.59484679</span>
<span class="co">#&gt; X_train_microD10S191 -1.30136747</span>
<span class="co">#&gt; X_train_microD4S394   1.34265359</span>
<span class="co">#&gt; X_train_microD1S197  -0.75014044</span>
<span class="co">#&gt; X_train_microD6S264   1.32746604</span>
<span class="co">#&gt; X_train_microD14S65  -3.20882866</span>
<span class="co">#&gt; X_train_microD17S790  0.55427680</span>
<span class="co">#&gt; X_train_microD5S430   3.40654627</span>
<span class="co">#&gt; X_train_microD3S1283  2.12510239</span>
<span class="co">#&gt; X_train_microD4S414   2.73619967</span>
<span class="co">#&gt; X_train_microD8S283   0.71955323</span>
<span class="co">#&gt; X_train_microD11S916  1.45026508</span>
<span class="co">#&gt; X_train_microD2S159   0.90293134</span>
<span class="co">#&gt; X_train_microD16S408 -0.59719901</span>
<span class="co">#&gt; X_train_microD6S275  -1.02204186</span>
<span class="co">#&gt; X_train_microD10S192  1.14220367</span>
<span class="co">#&gt; X_train_microsexe     0.67314561</span>
<span class="co">#&gt; X_train_microAgediag  0.04908478</span>
<span class="co">#&gt; X_train_microSiege   -0.41985924</span>
<span class="co">#&gt; X_train_microT        2.70581463</span>
<span class="co">#&gt; X_train_microN        2.47039973</span>
<span class="co">#&gt; X_train_microM       -4.53213922</span>
<span class="co">#&gt; X_train_microSTADE    0.48221697</span>
<span class="co">#&gt; Information criteria and Fit statistics:</span>
<span class="co">#&gt;                 AIC       BIC</span>
<span class="co">#&gt; Nb_Comp_0 112.87990 112.87990</span>
<span class="co">#&gt; Nb_Comp_1  85.11075  87.49278</span>
<span class="co">#&gt; Nb_Comp_2  75.49537  80.25942</span>
<span class="co">#&gt; Nb_Comp_3  68.45852  75.60460</span>
<span class="co">#&gt; Nb_Comp_4  63.09284  72.62094</span>
<span class="co">#&gt; Nb_Comp_5  55.30567  67.21581</span>

<span class="kw"><a href="reference/plsRcox.html">plsRcox</a></span>(<span class="dt">Xplan=</span>X_train_micro,<span class="dt">time=</span>Y_train_micro,<span class="dt">event=</span>C_train_micro,<span class="dt">nt=</span><span class="dv">5</span>,<span class="dt">sparse=</span><span class="ot">TRUE</span>, <span class="dt">alpha.pvals.expli=</span>.<span class="dv">15</span>)
<span class="co">#&gt; ____************************************************____</span>
<span class="co">#&gt; ____Component____ 1 ____</span>
<span class="co">#&gt; ____Component____ 2 ____</span>
<span class="co">#&gt; ____Component____ 3 ____</span>
<span class="co">#&gt; Warning : 25 &lt; 10^{-12}</span>
<span class="co">#&gt; Warning only 3 components could thus be extracted</span>
<span class="co">#&gt; ____Predicting X without NA neither in X nor in Y____</span>
<span class="co">#&gt; ****________________________________________________****</span>
<span class="co">#&gt; Number of required components:</span>
<span class="co">#&gt; [1] 5</span>
<span class="co">#&gt; Number of successfully computed components:</span>
<span class="co">#&gt; [1] 3</span>
<span class="co">#&gt; Coefficients:</span>
<span class="co">#&gt;                [,1]</span>
<span class="co">#&gt; D18S61   0.00000000</span>
<span class="co">#&gt; D17S794  0.00000000</span>
<span class="co">#&gt; D13S173  0.00000000</span>
<span class="co">#&gt; D20S107  2.22871454</span>
<span class="co">#&gt; TP53     0.00000000</span>
<span class="co">#&gt; D9S171   0.00000000</span>
<span class="co">#&gt; D8S264   0.00000000</span>
<span class="co">#&gt; D5S346  -1.20298526</span>
<span class="co">#&gt; D22S928  0.00000000</span>
<span class="co">#&gt; D18S53   0.00000000</span>
<span class="co">#&gt; D1S225  -1.29459798</span>
<span class="co">#&gt; D3S1282 -1.99426291</span>
<span class="co">#&gt; D15S127  1.39645601</span>
<span class="co">#&gt; D1S305   0.00000000</span>
<span class="co">#&gt; D1S207   1.25164327</span>
<span class="co">#&gt; D2S138  -1.65740160</span>
<span class="co">#&gt; D16S422  0.00000000</span>
<span class="co">#&gt; D9S179   0.00000000</span>
<span class="co">#&gt; D10S191 -1.25360805</span>
<span class="co">#&gt; D4S394   0.00000000</span>
<span class="co">#&gt; D1S197   0.00000000</span>
<span class="co">#&gt; D6S264   0.00000000</span>
<span class="co">#&gt; D14S65  -1.33587373</span>
<span class="co">#&gt; D17S790  0.00000000</span>
<span class="co">#&gt; D5S430   1.72799213</span>
<span class="co">#&gt; D3S1283  0.00000000</span>
<span class="co">#&gt; D4S414   1.03558702</span>
<span class="co">#&gt; D8S283   0.00000000</span>
<span class="co">#&gt; D11S916  0.00000000</span>
<span class="co">#&gt; D2S159   0.00000000</span>
<span class="co">#&gt; D16S408 -1.75748257</span>
<span class="co">#&gt; D6S275   0.00000000</span>
<span class="co">#&gt; D10S192  0.00000000</span>
<span class="co">#&gt; sexe     0.00000000</span>
<span class="co">#&gt; Agediag  0.05075304</span>
<span class="co">#&gt; Siege    0.00000000</span>
<span class="co">#&gt; T        1.36569407</span>
<span class="co">#&gt; N        1.27485618</span>
<span class="co">#&gt; M       -1.17682617</span>
<span class="co">#&gt; STADE   -0.65106093</span>
<span class="co">#&gt; Information criteria and Fit statistics:</span>
<span class="co">#&gt;                 AIC       BIC</span>
<span class="co">#&gt; Nb_Comp_0 112.87990 112.87990</span>
<span class="co">#&gt; Nb_Comp_1  85.54313  87.92516</span>
<span class="co">#&gt; Nb_Comp_2  75.16125  79.92530</span>
<span class="co">#&gt; Nb_Comp_3  73.63097  80.77705</span>

<span class="kw"><a href="reference/plsRcox.html">plsRcox</a></span>(<span class="dt">Xplan=</span><span class="op">~</span>X_train_micro,<span class="dt">time=</span>Y_train_micro,<span class="dt">event=</span>C_train_micro,<span class="dt">nt=</span><span class="dv">5</span>,<span class="dt">sparse=</span><span class="ot">TRUE</span>, <span class="dt">alpha.pvals.expli=</span>.<span class="dv">15</span>)
<span class="co">#&gt; ____************************************************____</span>
<span class="co">#&gt; ____Component____ 1 ____</span>
<span class="co">#&gt; ____Component____ 2 ____</span>
<span class="co">#&gt; ____Component____ 3 ____</span>
<span class="co">#&gt; Warning : 25 &lt; 10^{-12}</span>
<span class="co">#&gt; Warning only 3 components could thus be extracted</span>
<span class="co">#&gt; ____Predicting X without NA neither in X nor in Y____</span>
<span class="co">#&gt; ****________________________________________________****</span>
<span class="co">#&gt; Number of required components:</span>
<span class="co">#&gt; [1] 5</span>
<span class="co">#&gt; Number of successfully computed components:</span>
<span class="co">#&gt; [1] 3</span>
<span class="co">#&gt; Coefficients:</span>
<span class="co">#&gt;                             [,1]</span>
<span class="co">#&gt; X_train_microD18S61   0.00000000</span>
<span class="co">#&gt; X_train_microD17S794  0.00000000</span>
<span class="co">#&gt; X_train_microD13S173  0.00000000</span>
<span class="co">#&gt; X_train_microD20S107  2.22871454</span>
<span class="co">#&gt; X_train_microTP53     0.00000000</span>
<span class="co">#&gt; X_train_microD9S171   0.00000000</span>
<span class="co">#&gt; X_train_microD8S264   0.00000000</span>
<span class="co">#&gt; X_train_microD5S346  -1.20298526</span>
<span class="co">#&gt; X_train_microD22S928  0.00000000</span>
<span class="co">#&gt; X_train_microD18S53   0.00000000</span>
<span class="co">#&gt; X_train_microD1S225  -1.29459798</span>
<span class="co">#&gt; X_train_microD3S1282 -1.99426291</span>
<span class="co">#&gt; X_train_microD15S127  1.39645601</span>
<span class="co">#&gt; X_train_microD1S305   0.00000000</span>
<span class="co">#&gt; X_train_microD1S207   1.25164327</span>
<span class="co">#&gt; X_train_microD2S138  -1.65740160</span>
<span class="co">#&gt; X_train_microD16S422  0.00000000</span>
<span class="co">#&gt; X_train_microD9S179   0.00000000</span>
<span class="co">#&gt; X_train_microD10S191 -1.25360805</span>
<span class="co">#&gt; X_train_microD4S394   0.00000000</span>
<span class="co">#&gt; X_train_microD1S197   0.00000000</span>
<span class="co">#&gt; X_train_microD6S264   0.00000000</span>
<span class="co">#&gt; X_train_microD14S65  -1.33587373</span>
<span class="co">#&gt; X_train_microD17S790  0.00000000</span>
<span class="co">#&gt; X_train_microD5S430   1.72799213</span>
<span class="co">#&gt; X_train_microD3S1283  0.00000000</span>
<span class="co">#&gt; X_train_microD4S414   1.03558702</span>
<span class="co">#&gt; X_train_microD8S283   0.00000000</span>
<span class="co">#&gt; X_train_microD11S916  0.00000000</span>
<span class="co">#&gt; X_train_microD2S159   0.00000000</span>
<span class="co">#&gt; X_train_microD16S408 -1.75748257</span>
<span class="co">#&gt; X_train_microD6S275   0.00000000</span>
<span class="co">#&gt; X_train_microD10S192  0.00000000</span>
<span class="co">#&gt; X_train_microsexe     0.00000000</span>
<span class="co">#&gt; X_train_microAgediag  0.05075304</span>
<span class="co">#&gt; X_train_microSiege    0.00000000</span>
<span class="co">#&gt; X_train_microT        1.36569407</span>
<span class="co">#&gt; X_train_microN        1.27485618</span>
<span class="co">#&gt; X_train_microM       -1.17682617</span>
<span class="co">#&gt; X_train_microSTADE   -0.65106093</span>
<span class="co">#&gt; Information criteria and Fit statistics:</span>
<span class="co">#&gt;                 AIC       BIC</span>
<span class="co">#&gt; Nb_Comp_0 112.87990 112.87990</span>
<span class="co">#&gt; Nb_Comp_1  85.54313  87.92516</span>
<span class="co">#&gt; Nb_Comp_2  75.16125  79.92530</span>
<span class="co">#&gt; Nb_Comp_3  73.63097  80.77705</span></code></pre></div>
<div id="cv-plsrcox" class="section level3">
<h3 class="hasAnchor">
<a href="#cv-plsrcox" class="anchor"></a>cv.plsRcox</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">123456</span>)

(<span class="dt">cv.plsRcox.res=</span><span class="kw"><a href="reference/cv.plsRcox.html">cv.plsRcox</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">x=</span>X_train_micro,<span class="dt">time=</span>Y_train_micro,<span class="dt">status=</span>C_train_micro),<span class="dt">nt=</span><span class="dv">10</span>,<span class="dt">verbose =</span> <span class="ot">FALSE</span>))</code></pre></div>
<p><img src="reference/figures/README-cvplsRcox-1.png" width="100%"></p>
<pre><code>#&gt; $nt
#&gt; [1] 10
#&gt; 
#&gt; $cv.error5
#&gt;  [1] 0.5000000 0.9680950 0.9682845 0.9678603 0.9577447 0.9674094 0.9685457
#&gt;  [8] 0.9671530 0.9654175 0.9600213 0.9648240
#&gt; 
#&gt; $cv.se5
#&gt;  [1] 5.551115e-18 3.934293e-04 6.577693e-04 2.996472e-03 1.929769e-03
#&gt;  [6] 1.046829e-03 1.425160e-03 9.134268e-04 3.557709e-03 2.614560e-03
#&gt; [11] 5.990163e-04
#&gt; 
#&gt; $folds
#&gt; $folds$`1`
#&gt;  [1] 64 15 56 59 10 39 70  4 80 16 37 33 50 51 78 44
#&gt; 
#&gt; $folds$`2`
#&gt;  [1] 60 40 41 57  5 61 43 42 29 20 32 68  1 36 55 17
#&gt; 
#&gt; $folds$`3`
#&gt;  [1] 31  8 62 13 73 47 71 46 35 69 75 26 74 65 58 63
#&gt; 
#&gt; $folds$`4`
#&gt;  [1] 27 72 79 21 67 45  6 30 11 22 24 53 23  2 49  7
#&gt; 
#&gt; $folds$`5`
#&gt;  [1] 28 12 66 48 77  9 38 25 19 54 34 52  3 14 76 18
#&gt; 
#&gt; 
#&gt; $lambda.min5
#&gt; [1] 6
#&gt; 
#&gt; $lambda.1se5
#&gt; [1] 0</code></pre>
</div>
</div>
<div id="dkplsrcox" class="section level2">
<h2 class="hasAnchor">
<a href="#dkplsrcox" class="anchor"></a>DKplsRcox</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/DKplsRcox.html">DKplsRcox</a></span>(X_train_micro,<span class="dt">time=</span>Y_train_micro,<span class="dt">event=</span>C_train_micro,<span class="dt">nt=</span><span class="dv">5</span>)
<span class="co">#&gt; Kernel :  rbfdot </span>
<span class="co">#&gt; Estimated_sigma  0.01224473 </span>
<span class="co">#&gt; ____************************************************____</span>
<span class="co">#&gt; ____Component____ 1 ____</span>
<span class="co">#&gt; ____Component____ 2 ____</span>
<span class="co">#&gt; ____Component____ 3 ____</span>
<span class="co">#&gt; ____Component____ 4 ____</span>
<span class="co">#&gt; ____Component____ 5 ____</span>
<span class="co">#&gt; ____Predicting X without NA neither in X nor in Y____</span>
<span class="co">#&gt; ****________________________________________________****</span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; coxph(formula = YCsurv ~ ., data = tt_DKplsRcox)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;              coef  exp(coef)   se(coef)      z        p</span>
<span class="co">#&gt; Comp_1 -8.697e+00  1.670e-04  3.130e+00 -2.779 0.005450</span>
<span class="co">#&gt; Comp_2  2.689e+01  4.753e+11  7.117e+00  3.778 0.000158</span>
<span class="co">#&gt; Comp_3  2.097e+01  1.283e+09  5.799e+00  3.617 0.000299</span>
<span class="co">#&gt; Comp_4  9.230e+00  1.020e+04  3.050e+00  3.026 0.002475</span>
<span class="co">#&gt; Comp_5  8.445e+00  4.653e+03  2.626e+00  3.216 0.001302</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Likelihood ratio test=69.98  on 5 df, p=1.033e-13</span>
<span class="co">#&gt; n= 80, number of events= 17</span>

<span class="kw"><a href="reference/DKplsRcox.html">DKplsRcox</a></span>(<span class="op">~</span>X_train_micro,<span class="dt">time=</span>Y_train_micro,<span class="dt">event=</span>C_train_micro,<span class="dt">nt=</span><span class="dv">5</span>)
<span class="co">#&gt; Kernel :  rbfdot </span>
<span class="co">#&gt; Estimated_sigma  0.01198748 </span>
<span class="co">#&gt; ____************************************************____</span>
<span class="co">#&gt; ____Component____ 1 ____</span>
<span class="co">#&gt; ____Component____ 2 ____</span>
<span class="co">#&gt; ____Component____ 3 ____</span>
<span class="co">#&gt; ____Component____ 4 ____</span>
<span class="co">#&gt; ____Component____ 5 ____</span>
<span class="co">#&gt; ____Predicting X without NA neither in X nor in Y____</span>
<span class="co">#&gt; ****________________________________________________****</span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; coxph(formula = YCsurv ~ ., data = tt_DKplsRcox)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;              coef  exp(coef)   se(coef)      z        p</span>
<span class="co">#&gt; Comp_1 -9.302e+00  9.124e-05  3.258e+00 -2.855 0.004307</span>
<span class="co">#&gt; Comp_2  2.708e+01  5.780e+11  7.138e+00  3.794 0.000148</span>
<span class="co">#&gt; Comp_3  2.083e+01  1.108e+09  5.742e+00  3.627 0.000287</span>
<span class="co">#&gt; Comp_4  8.959e+00  7.774e+03  2.982e+00  3.004 0.002663</span>
<span class="co">#&gt; Comp_5  8.369e+00  4.311e+03  2.610e+00  3.207 0.001343</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Likelihood ratio test=69.46  on 5 df, p=1.328e-13</span>
<span class="co">#&gt; n= 80, number of events= 17</span>

<span class="kw"><a href="reference/DKplsRcox.html">DKplsRcox</a></span>(<span class="dt">Xplan=</span>X_train_micro,<span class="dt">time=</span>Y_train_micro,<span class="dt">event=</span>C_train_micro,<span class="dt">nt=</span><span class="dv">5</span>,<span class="dt">sparse=</span><span class="ot">TRUE</span>, <span class="dt">alpha.pvals.expli=</span>.<span class="dv">15</span>)
<span class="co">#&gt; Kernel :  rbfdot </span>
<span class="co">#&gt; Estimated_sigma  0.0131134 </span>
<span class="co">#&gt; ____************************************************____</span>
<span class="co">#&gt; ____Component____ 1 ____</span>
<span class="co">#&gt; ____Component____ 2 ____</span>
<span class="co">#&gt; ____Component____ 3 ____</span>
<span class="co">#&gt; ____Component____ 4 ____</span>
<span class="co">#&gt; ____Component____ 5 ____</span>
<span class="co">#&gt; ____Predicting X without NA neither in X nor in Y____</span>
<span class="co">#&gt; ****________________________________________________****</span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; coxph(formula = YCsurv ~ ., data = tt_DKplsRcox)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;              coef  exp(coef)   se(coef)      z        p</span>
<span class="co">#&gt; Comp_1 -6.302e+01  4.283e-28  1.917e+01 -3.288 0.001009</span>
<span class="co">#&gt; Comp_2  9.190e+01  8.127e+39  2.721e+01  3.377 0.000732</span>
<span class="co">#&gt; Comp_3  4.391e+01  1.179e+19  1.319e+01  3.330 0.000867</span>
<span class="co">#&gt; Comp_4  4.459e+01  2.320e+19  1.415e+01  3.151 0.001629</span>
<span class="co">#&gt; Comp_5  9.395e+00  1.202e+04  3.430e+00  2.739 0.006166</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Likelihood ratio test=81.58  on 5 df, p=3.921e-16</span>
<span class="co">#&gt; n= 80, number of events= 17</span>

<span class="kw"><a href="reference/DKplsRcox.html">DKplsRcox</a></span>(<span class="dt">Xplan=</span><span class="op">~</span>X_train_micro,<span class="dt">time=</span>Y_train_micro,<span class="dt">event=</span>C_train_micro,<span class="dt">nt=</span><span class="dv">5</span>,<span class="dt">sparse=</span><span class="ot">TRUE</span>, <span class="dt">alpha.pvals.expli=</span>.<span class="dv">15</span>)
<span class="co">#&gt; Kernel :  rbfdot </span>
<span class="co">#&gt; Estimated_sigma  0.01252998 </span>
<span class="co">#&gt; ____************************************************____</span>
<span class="co">#&gt; ____Component____ 1 ____</span>
<span class="co">#&gt; ____Component____ 2 ____</span>
<span class="co">#&gt; ____Component____ 3 ____</span>
<span class="co">#&gt; ____Component____ 4 ____</span>
<span class="co">#&gt; ____Component____ 5 ____</span>
<span class="co">#&gt; ____Predicting X without NA neither in X nor in Y____</span>
<span class="co">#&gt; ****________________________________________________****</span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; coxph(formula = YCsurv ~ ., data = tt_DKplsRcox)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;              coef  exp(coef)   se(coef)      z       p</span>
<span class="co">#&gt; Comp_1 -7.380e+01  8.860e-33  2.455e+01 -3.006 0.00264</span>
<span class="co">#&gt; Comp_2  1.108e+02  1.327e+48  3.640e+01  3.044 0.00233</span>
<span class="co">#&gt; Comp_3  4.924e+01  2.437e+21  1.627e+01  3.026 0.00248</span>
<span class="co">#&gt; Comp_4  5.359e+01  1.882e+23  1.800e+01  2.977 0.00291</span>
<span class="co">#&gt; Comp_5  1.116e+01  7.038e+04  4.011e+00  2.783 0.00539</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Likelihood ratio test=83.79  on 5 df, p=&lt; 2.2e-16</span>
<span class="co">#&gt; n= 80, number of events= 17</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from CRAN at <br><a href="https://cloud.r-project.org/package=plsRcox">https://​cloud.r-project.org/​package=plsRcox</a>
</li>
<li>Browse source code at <br><a href="https://github.com/fbertran/plsRcox">https://​github.com/​fbertran/​plsRcox</a>
</li>
<li>Report a bug at <br><a href="https://github.com/fbertran/plsRcox/issues">https://​github.com/​fbertran/​plsRcox/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing plsRcox</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Frederic Bertrand <br><small class="roles"> Maintainer, author </small> <a href="https://orcid.org/0000-0002-0837-8281" target="orcid.widget"><img src="https://members.orcid.org/sites/default/files/vector_iD_icon.svg" class="orcid" alt="ORCID" height="16"></a> </li>
<li>Myriam Maumy-Bertrand <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-4615-1512" target="orcid.widget"><img src="https://members.orcid.org/sites/default/files/vector_iD_icon.svg" class="orcid" alt="ORCID" height="16"></a> </li>
</ul>
</div>

      <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://cran.r-project.org/package=plsRcox"><img src="https://www.r-pkg.org/badges/version/plsRcox" alt="CRAN status"></a></li>
</ul>
</div>
</div>

</div>


      <footer><div class="copyright">
  <p>Developed by Frederic Bertrand, Myriam Maumy-Bertrand.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
