<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Partial Least Squares Regression for Cox Models and Related Techniques • plsRcox</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="apple-touch-icon.png">
<link rel="icon" sizes="any" href="favicon.ico">
<link rel="manifest" href="site.webmanifest">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Partial Least Squares Regression for Cox Models and Related Techniques">
<meta name="description" content="Provides Partial least squares Regression and various regular, sparse or kernel, techniques for fitting Cox models in high dimensional settings &lt;doi:10.1093/bioinformatics/btu660&gt;, Bastien, P., Bertrand, F., Meyer N., Maumy-Bertrand, M. (2015), Deviance residuals-based sparse PLS and sparse kernel PLS regression for censored data, Bioinformatics, 31(3):397-404. Cross validation criteria were studied in &lt;doi:10.48550/arXiv.1810.02962&gt;, Bertrand, F., Bastien, Ph. and Maumy-Bertrand, M. (2018), Cross validating extensions of kernel, sparse or regular partial least squares regression models to censored data.">
<meta property="og:description" content="Provides Partial least squares Regression and various regular, sparse or kernel, techniques for fitting Cox models in high dimensional settings &lt;doi:10.1093/bioinformatics/btu660&gt;, Bastien, P., Bertrand, F., Meyer N., Maumy-Bertrand, M. (2015), Deviance residuals-based sparse PLS and sparse kernel PLS regression for censored data, Bioinformatics, 31(3):397-404. Cross validation criteria were studied in &lt;doi:10.48550/arXiv.1810.02962&gt;, Bertrand, F., Bastien, Ph. and Maumy-Bertrand, M. (2018), Cross validating extensions of kernel, sparse or regular partial least squares regression models to censored data.">
<meta property="og:image" content="https://fbertran.github.io/plsRcox/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">plsRcox</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.8.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/fbertran/plsRcox/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="plsrcox-">plsRcox <a class="anchor" aria-label="anchor" href="#plsrcox-"></a>
</h1>
</div>
</div>
<div class="section level1">
<h1 id="plsrcox-cox-models-in-a-high-dimensional-setting-in-r">plsRcox, Cox-Models in a High Dimensional Setting in R<a class="anchor" aria-label="anchor" href="#plsrcox-cox-models-in-a-high-dimensional-setting-in-r"></a>
</h1>
<div class="section level2">
<h2 id="frédéric-bertrand-and-myriam-maumy-bertrand">Frédéric Bertrand and Myriam Maumy-Bertrand<a class="anchor" aria-label="anchor" href="#fr%C3%A9d%C3%A9ric-bertrand-and-myriam-maumy-bertrand"></a>
</h2>
<p><a href="https://doi.org/10.32614/CRAN.package.plsRcox" class="external-link uri">https://doi.org/10.32614/CRAN.package.plsRcox</a></p>
<!-- badges: start -->

<p>The goal of plsRcox is provide Cox models in a high dimensional setting in R.</p>
<p>plsRcox implements partial least squares Regression and various regular, sparse or kernel, techniques for fitting Cox models in high dimensional settings <a href="https://doi.org/10.1093/bioinformatics/btu660" class="external-link uri">https://doi.org/10.1093/bioinformatics/btu660</a>, Bastien, P., Bertrand, F., Meyer N., Maumy-Bertrand, M. (2015), Deviance residuals-based sparse PLS and sparse kernel PLS regression for censored data, Bioinformatics, 31(3):397-404. Cross validation criteria were studied in <a href="https://arxiv.org/abs/1810.01005" class="external-link">&lt;arXiv:1810.02962&gt;</a>, Bertrand, F., Bastien, Ph. and Maumy-Bertrand, M. (2018), Cross validating extensions of kernel, sparse or regular partial least squares regression models to censored data.</p>
<p>The package was presented at the <a href="https://user2014.r-project.org/" class="external-link">User2014!</a> conference. Frédéric Bertrand, Philippe Bastien, Nicolas Meyer and Myriam Bertrand (2014). “plsRcox, Cox-Models in a high dimensional setting in R”, <em>book of abstracts</em>, User2014!, Los Angeles, page 177, <a href="https://user2014.r-project.org/abstracts/posters/177_Bertrand.pdf" class="external-link uri">https://user2014.r-project.org/abstracts/posters/177_Bertrand.pdf</a>.</p>
<p>The plsRcox package contains an original allelotyping dataset from “Allelotyping identification of genomic alterations in rectal chromosomally unstable tumors without preoperative treatment”, Benoît Romain, Agnès Neuville, Nicolas Meyer, Cécile Brigand, Serge Rohr, Anne Schneider, Marie-Pierre Gaub and Dominique Guenot (2010), <em>BMC Cancer</em>, <strong>10</strong>:561, <a href="https://doi.org/10.1186/1471-2407-10-561" class="external-link uri">https://doi.org/10.1186/1471-2407-10-561</a>.</p>
<p>Support for parallel computation and GPU is being developped.</p>
<p>The package provides several modelling techniques related to penalized Cox models or extensions of partial least squares to Cox models. The first two were new algorithms.</p>
<ul>
<li><p><strong>coxsplsDR</strong> and <strong>cv.coxsplsDR</strong> (Philippe Bastien, Frederic Bertrand, Nicolas Meyer, and Myriam Maumy-Bertrand (2015), “Deviance residuals-based sparse PLS and sparse kernel PLS regression for censored data”, <em>Bioinformatics</em>, <strong>31</strong>(3):397-404, <a href="https://doi.org/10.1093/bioinformatics/btu660" class="external-link uri">https://doi.org/10.1093/bioinformatics/btu660</a>),</p></li>
<li><p><strong>coxDKsplsDR</strong> and <strong>cv.coxDKsplsDR</strong> (Philippe Bastien, Frederic Bertrand, Nicolas Meyer, and Myriam Maumy-Bertrand (2015), “Deviance residuals-based sparse PLS and sparse kernel PLS regression for censored data”, <em>Bioinformatics</em>, <strong>31</strong>(3):397-404, <a href="https://doi.org/10.1093/bioinformatics/btu660" class="external-link uri">https://doi.org/10.1093/bioinformatics/btu660</a>),</p></li>
<li><p><strong>coxDKplsDR</strong> and <strong>cv.coxDKplsDR</strong> (Philippe Bastien (2008), “Deviance residuals based PLS regression for censored data in high dimensional setting”, <em>Chemometrics and Intelligent Laboratory Systems</em>, <strong>91</strong>:78–86, <a href="https://doi.org/10.1016/j.chemolab.2007.09.009" class="external-link uri">https://doi.org/10.1016/j.chemolab.2007.09.009</a>),</p></li>
<li><p><strong>coxpls</strong> and <strong>cv.coxpls</strong> (Nguyen, D.V., Rocke, D.M. (2002), “Partial least squares proportional hazard regression for application to DNA microarray survival data”, <em>Bioinformatics</em>, <strong>18</strong>(12):1625–1632),</p></li>
<li><p><strong>coxplsDR</strong> and <strong>cv.coxplsDR</strong> (Philippe Bastien (2008), “Deviance residuals based PLS regression for censored data in high dimensional setting”, <em>Chemometrics and Intelligent Laboratory Systems</em>, <strong>91</strong>:78–86, <a href="https://doi.org/10.1016/j.chemolab.2007.09.009" class="external-link uri">https://doi.org/10.1016/j.chemolab.2007.09.009</a>),</p></li>
<li><p><strong>DKplsRcox</strong>,</p></li>
<li><p><strong>larsDR</strong> and <strong>cv.larsDR</strong> (Segal, M.R. (2006), “Microarray Gene Expression Data with Linked Survival Phenotypes: Diffuse large-B- Cell Lymphoma Revisited”, <em>Biostatistics</em>, <strong>7</strong>:268-285, <a href="https://doi.org/10.1093/biostatistics/kxj006" class="external-link uri">https://doi.org/10.1093/biostatistics/kxj006</a>),</p></li>
<li><p><strong>plsRcox</strong> and <strong>cv.plsRcox</strong> (Philippe Bastien, Vincenzo Esposito Vinzi, and Michel Tenenhaus (2005), “PLS generalised linear regression”, <em>Computational Statistics &amp; Data Analysis</em>, <strong>48</strong>(1):17–46, <a href="https://doi.org/10.1016/j.csda.2004.02.005" class="external-link uri">https://doi.org/10.1016/j.csda.2004.02.005</a>),</p></li>
<li><p><strong>autoplsRcox</strong> and <strong>cv.autoplsRcox</strong> (Philippe Bastien, Vincenzo Esposito Vinzi, and Michel Tenenhaus (2005), “PLS generalised linear regression”, <em>Computational Statistics &amp; Data Analysis</em>, <strong>48</strong>(1):17–46, <a href="https://doi.org/10.1016/j.csda.2004.02.005" class="external-link uri">https://doi.org/10.1016/j.csda.2004.02.005</a>),</p></li>
</ul>
<div class="float">
<img src="reference/figures/boxplots_bioinfo.png" alt="Performance comparisons."><div class="figcaption">Performance comparisons.</div>
</div>
<div class="float">
<img src="reference/figures/biplot_bioinfo.png" alt="Example of biplot of data."><div class="figcaption">Example of biplot of data.</div>
</div>
<p>This website and these examples were created by F. Bertrand and M. Maumy-Bertrand.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the released version of plsRcox from <a href="https://CRAN.R-project.org" class="external-link">CRAN</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"plsRcox"</span><span class="op">)</span></span></code></pre></div>
<p>You can install the development version of plsRcox from <a href="https://github.com" class="external-link">github</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"fbertran/plsRcox"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<div class="section level3">
<h3 id="the-original-allelotyping-dataset">The original allelotyping dataset<a class="anchor" aria-label="anchor" href="#the-original-allelotyping-dataset"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://fbertran.github.io/plsRcox/">plsRcox</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">micro.censure</span><span class="op">)</span></span>
<span><span class="va">Y_train_micro</span> <span class="op">&lt;-</span> <span class="va">micro.censure</span><span class="op">$</span><span class="va">survyear</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">80</span><span class="op">]</span></span>
<span><span class="va">C_train_micro</span> <span class="op">&lt;-</span> <span class="va">micro.censure</span><span class="op">$</span><span class="va">DC</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">80</span><span class="op">]</span></span>
<span><span class="va">Y_test_micro</span> <span class="op">&lt;-</span> <span class="va">micro.censure</span><span class="op">$</span><span class="va">survyear</span><span class="op">[</span><span class="fl">81</span><span class="op">:</span><span class="fl">117</span><span class="op">]</span></span>
<span><span class="va">C_test_micro</span> <span class="op">&lt;-</span> <span class="va">micro.censure</span><span class="op">$</span><span class="va">DC</span><span class="op">[</span><span class="fl">81</span><span class="op">:</span><span class="fl">117</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Xmicro.censure_compl_imp</span><span class="op">)</span></span>
<span><span class="va">X_train_micro</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">Xmicro.censure_compl_imp</span><span class="op">)</span><span class="op">)</span>,FUN<span class="op">=</span><span class="st">"as.numeric"</span>,MARGIN<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">80</span>,<span class="op">]</span></span>
<span><span class="va">X_train_micro_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">X_train_micro</span><span class="op">)</span></span></code></pre></div>
<p>Compute deviance residuals with some options.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/DR_coxph.html">DR_coxph</a></span><span class="op">(</span><span class="va">Y_train_micro</span>,<span class="va">C_train_micro</span>,plot<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="reference/figures/README-devianceresiduals-1.png" alt="plot of chunk devianceresiduals" width="100%"><p class="caption">
plot of chunk devianceresiduals
</p>
</div>
<pre><code><span><span class="co">#&gt;           1           2           3           4           5           6</span></span>
<span><span class="co">#&gt; -1.48432960 -0.54695398 -0.23145502 -0.34003013 -0.97633722 -0.38667660</span></span>
<span><span class="co">#&gt;           7           8           9          10          11          12</span></span>
<span><span class="co">#&gt; -0.38667660  1.57418914 -0.54695398 -0.15811388  2.10405254 -0.23145502</span></span>
<span><span class="co">#&gt;          13          14          15          16          17          18</span></span>
<span><span class="co">#&gt; -0.38667660 -1.09692040 -0.15811388 -0.15811388 -0.54695398 -0.38667660</span></span>
<span><span class="co">#&gt;          19          20          21          22          23          24</span></span>
<span><span class="co">#&gt;  0.65978609 -1.09692040 -0.43627414 -0.28961087 -0.38667660 -0.97633722</span></span>
<span><span class="co">#&gt;          25          26          27          28          29          30</span></span>
<span><span class="co">#&gt; -1.09692040 -0.15811388 -0.43627414 -0.43627414 -0.38667660 -0.23145502</span></span>
<span><span class="co">#&gt;          31          32          33          34          35          36</span></span>
<span><span class="co">#&gt;  2.30072697 -0.49023986 -0.54695398 -0.73444882  1.31082939 -0.97633722</span></span>
<span><span class="co">#&gt;          37          38          39          40          41          42</span></span>
<span><span class="co">#&gt;  1.70134282 -0.54695398 -0.15811388  1.07714870 -0.15811388 -0.49023986</span></span>
<span><span class="co">#&gt;          43          44          45          46          47          48</span></span>
<span><span class="co">#&gt; -0.34003013 -0.97633722 -0.15811388 -0.91410465 -1.09692040 -0.43627414</span></span>
<span><span class="co">#&gt;          49          50          51          52          53          54</span></span>
<span><span class="co">#&gt; -0.38667660 -0.09836581 -0.79392956  0.46851068 -0.34003013  1.95366297</span></span>
<span><span class="co">#&gt;          55          56          57          58          59          60</span></span>
<span><span class="co">#&gt;  2.60558118 -0.54695398 -1.09692040 -0.15811388 -0.49023986 -0.97633722</span></span>
<span><span class="co">#&gt;          61          62          63          64          65          66</span></span>
<span><span class="co">#&gt; -0.28961087  1.44879795  1.82660327 -0.38667660  0.96936094 -0.15811388</span></span>
<span><span class="co">#&gt;          67          68          69          70          71          72</span></span>
<span><span class="co">#&gt; -0.43627414 -0.49023986  1.18850436 -0.97633722 -0.97633722  0.86322194</span></span>
<span><span class="co">#&gt;          73          74          75          76          77          78</span></span>
<span><span class="co">#&gt; -0.43627414 -0.49023986 -0.38667660  0.76231394 -0.97633722 -0.43627414</span></span>
<span><span class="co">#&gt;          79          80</span></span>
<span><span class="co">#&gt; -0.54695398 -0.43627414</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/DR_coxph.html">DR_coxph</a></span><span class="op">(</span><span class="va">Y_train_micro</span>,<span class="va">C_train_micro</span>,scaleY<span class="op">=</span><span class="cn">FALSE</span>,plot<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="reference/figures/README-devianceresiduals2-1.png" alt="plot of chunk devianceresiduals2" width="100%"><p class="caption">
plot of chunk devianceresiduals2
</p>
</div>
<pre><code><span><span class="co">#&gt;           1           2           3           4           5           6</span></span>
<span><span class="co">#&gt; -1.48432960 -0.54695398 -0.23145502 -0.34003013 -0.97633722 -0.38667660</span></span>
<span><span class="co">#&gt;           7           8           9          10          11          12</span></span>
<span><span class="co">#&gt; -0.38667660  1.57418914 -0.54695398 -0.15811388  2.10405254 -0.23145502</span></span>
<span><span class="co">#&gt;          13          14          15          16          17          18</span></span>
<span><span class="co">#&gt; -0.38667660 -1.09692040 -0.15811388 -0.15811388 -0.54695398 -0.38667660</span></span>
<span><span class="co">#&gt;          19          20          21          22          23          24</span></span>
<span><span class="co">#&gt;  0.65978609 -1.09692040 -0.43627414 -0.28961087 -0.38667660 -0.97633722</span></span>
<span><span class="co">#&gt;          25          26          27          28          29          30</span></span>
<span><span class="co">#&gt; -1.09692040 -0.15811388 -0.43627414 -0.43627414 -0.38667660 -0.23145502</span></span>
<span><span class="co">#&gt;          31          32          33          34          35          36</span></span>
<span><span class="co">#&gt;  2.30072697 -0.49023986 -0.54695398 -0.73444882  1.31082939 -0.97633722</span></span>
<span><span class="co">#&gt;          37          38          39          40          41          42</span></span>
<span><span class="co">#&gt;  1.70134282 -0.54695398 -0.15811388  1.07714870 -0.15811388 -0.49023986</span></span>
<span><span class="co">#&gt;          43          44          45          46          47          48</span></span>
<span><span class="co">#&gt; -0.34003013 -0.97633722 -0.15811388 -0.91410465 -1.09692040 -0.43627414</span></span>
<span><span class="co">#&gt;          49          50          51          52          53          54</span></span>
<span><span class="co">#&gt; -0.38667660 -0.09836581 -0.79392956  0.46851068 -0.34003013  1.95366297</span></span>
<span><span class="co">#&gt;          55          56          57          58          59          60</span></span>
<span><span class="co">#&gt;  2.60558118 -0.54695398 -1.09692040 -0.15811388 -0.49023986 -0.97633722</span></span>
<span><span class="co">#&gt;          61          62          63          64          65          66</span></span>
<span><span class="co">#&gt; -0.28961087  1.44879795  1.82660327 -0.38667660  0.96936094 -0.15811388</span></span>
<span><span class="co">#&gt;          67          68          69          70          71          72</span></span>
<span><span class="co">#&gt; -0.43627414 -0.49023986  1.18850436 -0.97633722 -0.97633722  0.86322194</span></span>
<span><span class="co">#&gt;          73          74          75          76          77          78</span></span>
<span><span class="co">#&gt; -0.43627414 -0.49023986 -0.38667660  0.76231394 -0.97633722 -0.43627414</span></span>
<span><span class="co">#&gt;          79          80</span></span>
<span><span class="co">#&gt; -0.54695398 -0.43627414</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/DR_coxph.html">DR_coxph</a></span><span class="op">(</span><span class="va">Y_train_micro</span>,<span class="va">C_train_micro</span>,scaleY<span class="op">=</span><span class="cn">TRUE</span>,plot<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="reference/figures/README-devianceresiduals3-1.png" alt="plot of chunk devianceresiduals3" width="100%"><p class="caption">
plot of chunk devianceresiduals3
</p>
</div>
<pre><code><span><span class="co">#&gt;           1           2           3           4           5           6</span></span>
<span><span class="co">#&gt; -1.48432960 -0.54695398 -0.23145502 -0.34003013 -0.97633722 -0.38667660</span></span>
<span><span class="co">#&gt;           7           8           9          10          11          12</span></span>
<span><span class="co">#&gt; -0.38667660  1.57418914 -0.54695398 -0.15811388  2.10405254 -0.23145502</span></span>
<span><span class="co">#&gt;          13          14          15          16          17          18</span></span>
<span><span class="co">#&gt; -0.38667660 -1.09692040 -0.15811388 -0.15811388 -0.54695398 -0.38667660</span></span>
<span><span class="co">#&gt;          19          20          21          22          23          24</span></span>
<span><span class="co">#&gt;  0.65978609 -1.09692040 -0.43627414 -0.28961087 -0.38667660 -0.97633722</span></span>
<span><span class="co">#&gt;          25          26          27          28          29          30</span></span>
<span><span class="co">#&gt; -1.09692040 -0.15811388 -0.43627414 -0.43627414 -0.38667660 -0.23145502</span></span>
<span><span class="co">#&gt;          31          32          33          34          35          36</span></span>
<span><span class="co">#&gt;  2.30072697 -0.49023986 -0.54695398 -0.73444882  1.31082939 -0.97633722</span></span>
<span><span class="co">#&gt;          37          38          39          40          41          42</span></span>
<span><span class="co">#&gt;  1.70134282 -0.54695398 -0.15811388  1.07714870 -0.15811388 -0.49023986</span></span>
<span><span class="co">#&gt;          43          44          45          46          47          48</span></span>
<span><span class="co">#&gt; -0.34003013 -0.97633722 -0.15811388 -0.91410465 -1.09692040 -0.43627414</span></span>
<span><span class="co">#&gt;          49          50          51          52          53          54</span></span>
<span><span class="co">#&gt; -0.38667660 -0.09836581 -0.79392956  0.46851068 -0.34003013  1.95366297</span></span>
<span><span class="co">#&gt;          55          56          57          58          59          60</span></span>
<span><span class="co">#&gt;  2.60558118 -0.54695398 -1.09692040 -0.15811388 -0.49023986 -0.97633722</span></span>
<span><span class="co">#&gt;          61          62          63          64          65          66</span></span>
<span><span class="co">#&gt; -0.28961087  1.44879795  1.82660327 -0.38667660  0.96936094 -0.15811388</span></span>
<span><span class="co">#&gt;          67          68          69          70          71          72</span></span>
<span><span class="co">#&gt; -0.43627414 -0.49023986  1.18850436 -0.97633722 -0.97633722  0.86322194</span></span>
<span><span class="co">#&gt;          73          74          75          76          77          78</span></span>
<span><span class="co">#&gt; -0.43627414 -0.49023986 -0.38667660  0.76231394 -0.97633722 -0.43627414</span></span>
<span><span class="co">#&gt;          79          80</span></span>
<span><span class="co">#&gt; -0.54695398 -0.43627414</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="coxsplsdr">coxsplsDR<a class="anchor" aria-label="anchor" href="#coxsplsdr"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">cox_splsDR_fit</span><span class="op">=</span><span class="fu"><a href="reference/coxsplsDR.html">coxsplsDR</a></span><span class="op">(</span><span class="va">X_train_micro</span>,<span class="va">Y_train_micro</span>,<span class="va">C_train_micro</span>,ncomp<span class="op">=</span><span class="fl">6</span>,eta<span class="op">=</span><span class="fl">.5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Call:</span></span>
<span><span class="co">#&gt; coxph(formula = YCsurv ~ ., data = tt_splsDR)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         coef exp(coef) se(coef)     z        p</span></span>
<span><span class="co">#&gt; dim.1 0.8093    2.2462   0.2029 3.989 6.63e-05</span></span>
<span><span class="co">#&gt; dim.2 0.9295    2.5333   0.2939 3.163  0.00156</span></span>
<span><span class="co">#&gt; dim.3 0.9968    2.7096   0.4190 2.379  0.01736</span></span>
<span><span class="co">#&gt; dim.4 0.9705    2.6391   0.3793 2.558  0.01052</span></span>
<span><span class="co">#&gt; dim.5 0.2162    1.2413   0.2811 0.769  0.44192</span></span>
<span><span class="co">#&gt; dim.6 0.4380    1.5496   0.3608 1.214  0.22473</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Likelihood ratio test=55.06  on 6 df, p=4.51e-10</span></span>
<span><span class="co">#&gt; n= 80, number of events= 17</span></span>
<span></span>
<span><span class="op">(</span><span class="va">cox_splsDR_fit2</span><span class="op">=</span><span class="fu"><a href="reference/coxsplsDR.html">coxsplsDR</a></span><span class="op">(</span><span class="op">~</span><span class="va">X_train_micro</span>,<span class="va">Y_train_micro</span>,<span class="va">C_train_micro</span>,ncomp<span class="op">=</span><span class="fl">6</span>,eta<span class="op">=</span><span class="fl">.5</span>,trace<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in model.matrix(mt0, mf0, , contrasts.arg = contrasts.arg): l'argument "contrasts.arg" est manquant, avec aucune valeur par défaut</span></span>
<span></span>
<span><span class="op">(</span><span class="va">cox_splsDR_fit3</span><span class="op">=</span><span class="fu"><a href="reference/coxsplsDR.html">coxsplsDR</a></span><span class="op">(</span><span class="op">~</span><span class="va">.</span>,<span class="va">Y_train_micro</span>,<span class="va">C_train_micro</span>,ncomp<span class="op">=</span><span class="fl">6</span>,</span>
<span>dataXplan<span class="op">=</span><span class="va">X_train_micro_df</span>,eta<span class="op">=</span><span class="fl">.5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in model.matrix(mt0, mf0, , contrasts.arg = contrasts.arg): l'argument "contrasts.arg" est manquant, avec aucune valeur par défaut</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">cox_splsDR_fit</span>,<span class="va">cox_splsDR_fit2</span>,<span class="va">cox_splsDR_fit3</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in rm(cox_splsDR_fit, cox_splsDR_fit2, cox_splsDR_fit3): objet</span></span>
<span><span class="co">#&gt; 'cox_splsDR_fit2' introuvable</span></span>
<span><span class="co">#&gt; Warning in rm(cox_splsDR_fit, cox_splsDR_fit2, cox_splsDR_fit3): objet</span></span>
<span><span class="co">#&gt; 'cox_splsDR_fit3' introuvable</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cvcoxsplsdr">cv.coxsplsDR<a class="anchor" aria-label="anchor" href="#cvcoxsplsdr"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123456</span><span class="op">)</span></span>
<span></span>
<span><span class="op">(</span><span class="va">cv.coxsplsDR.res</span><span class="op">=</span><span class="fu"><a href="reference/cv.coxsplsDR.html">cv.coxsplsDR</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">X_train_micro</span>,time<span class="op">=</span><span class="va">Y_train_micro</span>, status<span class="op">=</span><span class="va">C_train_micro</span><span class="op">)</span>,nt<span class="op">=</span><span class="fl">10</span>,eta<span class="op">=</span><span class="fl">.1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; CV Fold 1 </span></span>
<span><span class="co">#&gt; CV Fold 2 </span></span>
<span><span class="co">#&gt; CV Fold 3 </span></span>
<span><span class="co">#&gt; CV Fold 4 </span></span>
<span><span class="co">#&gt; CV Fold 5</span></span></code></pre></div>
<p><img src="reference/figures/README-cvcoxsplsDR-1.png" title="plot of chunk cvcoxsplsDR" alt="plot of chunk cvcoxsplsDR" width="100%"></p>
<pre><code><span><span class="co">#&gt; $nt</span></span>
<span><span class="co">#&gt; [1] 10</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $cv.error10</span></span>
<span><span class="co">#&gt;  [1] 0.5000000 0.6786893 0.6913293 0.6485690 0.6656184 0.6591497 0.6733976</span></span>
<span><span class="co">#&gt;  [8] 0.6252317 0.6388320 0.6592004 0.6589521</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $cv.se10</span></span>
<span><span class="co">#&gt;  [1] 0.00000000 0.04017423 0.02726346 0.03897730 0.03874068 0.04042522 0.03952962</span></span>
<span><span class="co">#&gt;  [8] 0.04645295 0.04782038 0.05168926 0.05259748</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds</span></span>
<span><span class="co">#&gt; $folds$`1`</span></span>
<span><span class="co">#&gt;  [1] 60  3  2 14 77  6 50  4 72 32 22  1 41 21 63 25</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds$`2`</span></span>
<span><span class="co">#&gt;  [1] 42 67 65 15 73 48 57 26  7 13 31 53  5 27 37 64</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds$`3`</span></span>
<span><span class="co">#&gt;  [1] 71 23 56 35 75 29 30 18 62 44 12 33 68 49 43 55</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds$`4`</span></span>
<span><span class="co">#&gt;  [1] 54 76 24 16 34 66  9 11 69 40 70 36 39  8 19 20</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds$`5`</span></span>
<span><span class="co">#&gt;  [1] 74 38 46 80 47 78 10 45 51 28 61 79 58 17 52 59</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $lambda.min10</span></span>
<span><span class="co">#&gt; [1] 2</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $lambda.1se10</span></span>
<span><span class="co">#&gt; [1] 0</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $nzb</span></span>
<span><span class="co">#&gt;  [1]  0 34 40 40 40 40 40 40 40 40 40</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="coxdksplsdr">coxDKsplsDR<a class="anchor" aria-label="anchor" href="#coxdksplsdr"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">cox_DKsplsDR_fit</span><span class="op">=</span><span class="fu"><a href="reference/coxDKsplsDR.html">coxDKsplsDR</a></span><span class="op">(</span><span class="va">X_train_micro</span>,<span class="va">Y_train_micro</span>,<span class="va">C_train_micro</span>,ncomp<span class="op">=</span><span class="fl">6</span>, validation<span class="op">=</span><span class="st">"CV"</span>,eta<span class="op">=</span><span class="fl">.5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Kernel :  rbfdot </span></span>
<span><span class="co">#&gt; Estimated_sigma  0.0122308</span></span>
<span><span class="co">#&gt; Call:</span></span>
<span><span class="co">#&gt; coxph(formula = YCsurv ~ ., data = tt_DKsplsDR)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;            coef exp(coef)  se(coef)     z       p</span></span>
<span><span class="co">#&gt; dim.1 3.633e+00 3.783e+01 1.245e+00 2.918 0.00352</span></span>
<span><span class="co">#&gt; dim.2 9.905e+00 2.003e+04 3.297e+00 3.004 0.00266</span></span>
<span><span class="co">#&gt; dim.3 6.491e+00 6.589e+02 2.575e+00 2.521 0.01170</span></span>
<span><span class="co">#&gt; dim.4 1.465e+01 2.308e+06 4.848e+00 3.022 0.00251</span></span>
<span><span class="co">#&gt; dim.5 6.103e+00 4.473e+02 2.757e+00 2.213 0.02687</span></span>
<span><span class="co">#&gt; dim.6 1.249e+01 2.664e+05 4.980e+00 2.508 0.01212</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Likelihood ratio test=69.55  on 6 df, p=5.067e-13</span></span>
<span><span class="co">#&gt; n= 80, number of events= 17</span></span>
<span></span>
<span><span class="op">(</span><span class="va">cox_DKsplsDR_fit</span><span class="op">=</span><span class="fu"><a href="reference/coxDKsplsDR.html">coxDKsplsDR</a></span><span class="op">(</span><span class="op">~</span><span class="va">X_train_micro</span>,<span class="va">Y_train_micro</span>,<span class="va">C_train_micro</span>,ncomp<span class="op">=</span><span class="fl">6</span>, validation<span class="op">=</span><span class="st">"CV"</span>,eta<span class="op">=</span><span class="fl">.5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in model.matrix(mt0, mf0, , contrasts.arg = contrasts.arg): l'argument "contrasts.arg" est manquant, avec aucune valeur par défaut</span></span>
<span></span>
<span><span class="op">(</span><span class="va">cox_DKsplsDR_fit</span><span class="op">=</span><span class="fu"><a href="reference/coxDKsplsDR.html">coxDKsplsDR</a></span><span class="op">(</span><span class="op">~</span><span class="va">.</span>,<span class="va">Y_train_micro</span>,<span class="va">C_train_micro</span>,ncomp<span class="op">=</span><span class="fl">6</span>,</span>
<span>validation<span class="op">=</span><span class="st">"CV"</span>,dataXplan<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">X_train_micro</span><span class="op">)</span>,eta<span class="op">=</span><span class="fl">.5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in model.matrix(mt0, mf0, , contrasts.arg = contrasts.arg): l'argument "contrasts.arg" est manquant, avec aucune valeur par défaut</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">cox_DKsplsDR_fit</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cvcoxsplsdr-1">cv.coxsplsDR<a class="anchor" aria-label="anchor" href="#cvcoxsplsdr-1"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123456</span><span class="op">)</span></span>
<span></span>
<span><span class="op">(</span><span class="va">cv.coxDKsplsDR.res</span><span class="op">=</span><span class="fu"><a href="reference/cv.coxDKsplsDR.html">cv.coxDKsplsDR</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">X_train_micro</span>,time<span class="op">=</span><span class="va">Y_train_micro</span>, status<span class="op">=</span><span class="va">C_train_micro</span><span class="op">)</span>,nt<span class="op">=</span><span class="fl">10</span>,eta<span class="op">=</span><span class="fl">.1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Kernel :  rbfdot </span></span>
<span><span class="co">#&gt; Estimated_sigma  0.01257168 </span></span>
<span><span class="co">#&gt; CV Fold 1 </span></span>
<span><span class="co">#&gt; Kernel :  rbfdot </span></span>
<span><span class="co">#&gt; Estimated_sigma  0.01198263 </span></span>
<span><span class="co">#&gt; CV Fold 2 </span></span>
<span><span class="co">#&gt; Kernel :  rbfdot </span></span>
<span><span class="co">#&gt; Estimated_sigma  0.01156809 </span></span>
<span><span class="co">#&gt; CV Fold 3 </span></span>
<span><span class="co">#&gt; Kernel :  rbfdot </span></span>
<span><span class="co">#&gt; Estimated_sigma  0.01287851 </span></span>
<span><span class="co">#&gt; CV Fold 4 </span></span>
<span><span class="co">#&gt; Kernel :  rbfdot </span></span>
<span><span class="co">#&gt; Estimated_sigma  0.01127231 </span></span>
<span><span class="co">#&gt; CV Fold 5</span></span></code></pre></div>
<p><img src="reference/figures/README-cvcoxDKsplsDR-1.png" title="plot of chunk cvcoxDKsplsDR" alt="plot of chunk cvcoxDKsplsDR" width="100%"></p>
<pre><code><span><span class="co">#&gt; $nt</span></span>
<span><span class="co">#&gt; [1] 10</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $cv.error10</span></span>
<span><span class="co">#&gt;  [1] 0.5000000 0.6381540 0.6963262 0.6537039 0.6204813 0.6886401 0.6632860</span></span>
<span><span class="co">#&gt;  [8] 0.6349883 0.6762113 0.6261072 0.6087014</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $cv.se10</span></span>
<span><span class="co">#&gt;  [1] 0.00000000 0.03036225 0.02912723 0.04020941 0.03577022 0.03542745 0.03283778</span></span>
<span><span class="co">#&gt;  [8] 0.04532447 0.03390654 0.02968504 0.03306444</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds</span></span>
<span><span class="co">#&gt; $folds$`1`</span></span>
<span><span class="co">#&gt;  [1] 60  3  2 14 77  6 50  4 72 32 22  1 41 21 63 25</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds$`2`</span></span>
<span><span class="co">#&gt;  [1] 42 67 65 15 73 48 57 26  7 13 31 53  5 27 37 64</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds$`3`</span></span>
<span><span class="co">#&gt;  [1] 71 23 56 35 75 29 30 18 62 44 12 33 68 49 43 55</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds$`4`</span></span>
<span><span class="co">#&gt;  [1] 54 76 24 16 34 66  9 11 69 40 70 36 39  8 19 20</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds$`5`</span></span>
<span><span class="co">#&gt;  [1] 74 38 46 80 47 78 10 45 51 28 61 79 58 17 52 59</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $lambda.min10</span></span>
<span><span class="co">#&gt; [1] 2</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $lambda.1se10</span></span>
<span><span class="co">#&gt; [1] 0</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $nzb</span></span>
<span><span class="co">#&gt;  [1]  0 52 61 64 64 64 64 64 64 64 64</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="plsrcox">plsRcox<a class="anchor" aria-label="anchor" href="#plsrcox"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plsRcox.html">plsRcox</a></span><span class="op">(</span><span class="va">X_train_micro</span>,time<span class="op">=</span><span class="va">Y_train_micro</span>,event<span class="op">=</span><span class="va">C_train_micro</span>,nt<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; ____************************************************____</span></span>
<span><span class="co">#&gt; ____Component____ 1 ____</span></span>
<span><span class="co">#&gt; ____Component____ 2 ____</span></span>
<span><span class="co">#&gt; ____Component____ 3 ____</span></span>
<span><span class="co">#&gt; ____Component____ 4 ____</span></span>
<span><span class="co">#&gt; ____Component____ 5 ____</span></span>
<span><span class="co">#&gt; ____Predicting X without NA neither in X nor in Y____</span></span>
<span><span class="co">#&gt; ****________________________________________________****</span></span>
<span><span class="co">#&gt; Number of required components:</span></span>
<span><span class="co">#&gt; [1] 5</span></span>
<span><span class="co">#&gt; Number of successfully computed components:</span></span>
<span><span class="co">#&gt; [1] 5</span></span>
<span><span class="co">#&gt; Coefficients:</span></span>
<span><span class="co">#&gt;                [,1]</span></span>
<span><span class="co">#&gt; D18S61   0.68964919</span></span>
<span><span class="co">#&gt; D17S794 -1.14362392</span></span>
<span><span class="co">#&gt; D13S173  1.37632457</span></span>
<span><span class="co">#&gt; D20S107  4.96128745</span></span>
<span><span class="co">#&gt; TP53     1.68453950</span></span>
<span><span class="co">#&gt; D9S171  -1.46691252</span></span>
<span><span class="co">#&gt; D8S264   0.66710776</span></span>
<span><span class="co">#&gt; D5S346  -4.61338196</span></span>
<span><span class="co">#&gt; D22S928 -1.82005524</span></span>
<span><span class="co">#&gt; D18S53   0.79853646</span></span>
<span><span class="co">#&gt; D1S225  -1.46234986</span></span>
<span><span class="co">#&gt; D3S1282 -1.67925042</span></span>
<span><span class="co">#&gt; D15S127  3.92225537</span></span>
<span><span class="co">#&gt; D1S305  -2.29680161</span></span>
<span><span class="co">#&gt; D1S207   2.02539691</span></span>
<span><span class="co">#&gt; D2S138  -3.48975878</span></span>
<span><span class="co">#&gt; D16S422 -2.92189625</span></span>
<span><span class="co">#&gt; D9S179  -0.59484679</span></span>
<span><span class="co">#&gt; D10S191 -1.30136747</span></span>
<span><span class="co">#&gt; D4S394   1.34265359</span></span>
<span><span class="co">#&gt; D1S197  -0.75014044</span></span>
<span><span class="co">#&gt; D6S264   1.32746604</span></span>
<span><span class="co">#&gt; D14S65  -3.20882866</span></span>
<span><span class="co">#&gt; D17S790  0.55427680</span></span>
<span><span class="co">#&gt; D5S430   3.40654627</span></span>
<span><span class="co">#&gt; D3S1283  2.12510239</span></span>
<span><span class="co">#&gt; D4S414   2.73619967</span></span>
<span><span class="co">#&gt; D8S283   0.71955323</span></span>
<span><span class="co">#&gt; D11S916  1.45026508</span></span>
<span><span class="co">#&gt; D2S159   0.90293134</span></span>
<span><span class="co">#&gt; D16S408 -0.59719901</span></span>
<span><span class="co">#&gt; D6S275  -1.02204186</span></span>
<span><span class="co">#&gt; D10S192  1.14220367</span></span>
<span><span class="co">#&gt; sexe     0.67314561</span></span>
<span><span class="co">#&gt; Agediag  0.04908478</span></span>
<span><span class="co">#&gt; Siege   -0.41985924</span></span>
<span><span class="co">#&gt; T        2.70581463</span></span>
<span><span class="co">#&gt; N        2.47039973</span></span>
<span><span class="co">#&gt; M       -4.53213922</span></span>
<span><span class="co">#&gt; STADE    0.48221697</span></span>
<span><span class="co">#&gt; Information criteria and Fit statistics:</span></span>
<span><span class="co">#&gt;                 AIC       BIC</span></span>
<span><span class="co">#&gt; Nb_Comp_0 112.87990 112.87990</span></span>
<span><span class="co">#&gt; Nb_Comp_1  85.11075  87.49278</span></span>
<span><span class="co">#&gt; Nb_Comp_2  75.49537  80.25942</span></span>
<span><span class="co">#&gt; Nb_Comp_3  68.45852  75.60460</span></span>
<span><span class="co">#&gt; Nb_Comp_4  63.09284  72.62094</span></span>
<span><span class="co">#&gt; Nb_Comp_5  55.30567  67.21581</span></span>
<span></span>
<span><span class="fu"><a href="reference/plsRcox.html">plsRcox</a></span><span class="op">(</span><span class="op">~</span><span class="va">X_train_micro</span>,time<span class="op">=</span><span class="va">Y_train_micro</span>,event<span class="op">=</span><span class="va">C_train_micro</span>,nt<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in model.matrix(mt0, mf0, contrasts.arg = contrasts.arg): l'argument "contrasts.arg" est manquant, avec aucune valeur par défaut</span></span>
<span></span>
<span><span class="fu"><a href="reference/plsRcox.html">plsRcox</a></span><span class="op">(</span>Xplan<span class="op">=</span><span class="va">X_train_micro</span>,time<span class="op">=</span><span class="va">Y_train_micro</span>,event<span class="op">=</span><span class="va">C_train_micro</span>,nt<span class="op">=</span><span class="fl">5</span>,sparse<span class="op">=</span><span class="cn">TRUE</span>, alpha.pvals.expli<span class="op">=</span><span class="fl">.15</span><span class="op">)</span></span>
<span><span class="co">#&gt; ____************************************************____</span></span>
<span><span class="co">#&gt; ____Component____ 1 ____</span></span>
<span><span class="co">#&gt; ____Component____ 2 ____</span></span>
<span><span class="co">#&gt; ____Component____ 3 ____</span></span>
<span><span class="co">#&gt; Warning : 25 &lt; 10^{-12}</span></span>
<span><span class="co">#&gt; Warning only 3 components could thus be extracted</span></span>
<span><span class="co">#&gt; ____Predicting X without NA neither in X nor in Y____</span></span>
<span><span class="co">#&gt; ****________________________________________________****</span></span>
<span><span class="co">#&gt; Number of required components:</span></span>
<span><span class="co">#&gt; [1] 5</span></span>
<span><span class="co">#&gt; Number of successfully computed components:</span></span>
<span><span class="co">#&gt; [1] 3</span></span>
<span><span class="co">#&gt; Coefficients:</span></span>
<span><span class="co">#&gt;                [,1]</span></span>
<span><span class="co">#&gt; D18S61   0.00000000</span></span>
<span><span class="co">#&gt; D17S794  0.00000000</span></span>
<span><span class="co">#&gt; D13S173  0.00000000</span></span>
<span><span class="co">#&gt; D20S107  2.22871454</span></span>
<span><span class="co">#&gt; TP53     0.00000000</span></span>
<span><span class="co">#&gt; D9S171   0.00000000</span></span>
<span><span class="co">#&gt; D8S264   0.00000000</span></span>
<span><span class="co">#&gt; D5S346  -1.20298526</span></span>
<span><span class="co">#&gt; D22S928  0.00000000</span></span>
<span><span class="co">#&gt; D18S53   0.00000000</span></span>
<span><span class="co">#&gt; D1S225  -1.29459798</span></span>
<span><span class="co">#&gt; D3S1282 -1.99426291</span></span>
<span><span class="co">#&gt; D15S127  1.39645601</span></span>
<span><span class="co">#&gt; D1S305   0.00000000</span></span>
<span><span class="co">#&gt; D1S207   1.25164327</span></span>
<span><span class="co">#&gt; D2S138  -1.65740160</span></span>
<span><span class="co">#&gt; D16S422  0.00000000</span></span>
<span><span class="co">#&gt; D9S179   0.00000000</span></span>
<span><span class="co">#&gt; D10S191 -1.25360805</span></span>
<span><span class="co">#&gt; D4S394   0.00000000</span></span>
<span><span class="co">#&gt; D1S197   0.00000000</span></span>
<span><span class="co">#&gt; D6S264   0.00000000</span></span>
<span><span class="co">#&gt; D14S65  -1.33587373</span></span>
<span><span class="co">#&gt; D17S790  0.00000000</span></span>
<span><span class="co">#&gt; D5S430   1.72799213</span></span>
<span><span class="co">#&gt; D3S1283  0.00000000</span></span>
<span><span class="co">#&gt; D4S414   1.03558702</span></span>
<span><span class="co">#&gt; D8S283   0.00000000</span></span>
<span><span class="co">#&gt; D11S916  0.00000000</span></span>
<span><span class="co">#&gt; D2S159   0.00000000</span></span>
<span><span class="co">#&gt; D16S408 -1.75748257</span></span>
<span><span class="co">#&gt; D6S275   0.00000000</span></span>
<span><span class="co">#&gt; D10S192  0.00000000</span></span>
<span><span class="co">#&gt; sexe     0.00000000</span></span>
<span><span class="co">#&gt; Agediag  0.05075304</span></span>
<span><span class="co">#&gt; Siege    0.00000000</span></span>
<span><span class="co">#&gt; T        1.36569407</span></span>
<span><span class="co">#&gt; N        1.27485618</span></span>
<span><span class="co">#&gt; M       -1.17682617</span></span>
<span><span class="co">#&gt; STADE   -0.65106093</span></span>
<span><span class="co">#&gt; Information criteria and Fit statistics:</span></span>
<span><span class="co">#&gt;                 AIC       BIC</span></span>
<span><span class="co">#&gt; Nb_Comp_0 112.87990 112.87990</span></span>
<span><span class="co">#&gt; Nb_Comp_1  85.54313  87.92516</span></span>
<span><span class="co">#&gt; Nb_Comp_2  75.16125  79.92530</span></span>
<span><span class="co">#&gt; Nb_Comp_3  73.63097  80.77705</span></span>
<span></span>
<span><span class="fu"><a href="reference/plsRcox.html">plsRcox</a></span><span class="op">(</span>Xplan<span class="op">=</span><span class="op">~</span><span class="va">X_train_micro</span>,time<span class="op">=</span><span class="va">Y_train_micro</span>,event<span class="op">=</span><span class="va">C_train_micro</span>,nt<span class="op">=</span><span class="fl">5</span>,sparse<span class="op">=</span><span class="cn">TRUE</span>, alpha.pvals.expli<span class="op">=</span><span class="fl">.15</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in model.matrix(mt0, mf0, contrasts.arg = contrasts.arg): l'argument "contrasts.arg" est manquant, avec aucune valeur par défaut</span></span></code></pre></div>
<div class="section level3">
<h3 id="cvplsrcox">cv.plsRcox<a class="anchor" aria-label="anchor" href="#cvplsrcox"></a>
</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123456</span><span class="op">)</span></span>
<span></span>
<span><span class="op">(</span><span class="va">cv.plsRcox.res</span><span class="op">=</span><span class="fu"><a href="reference/cv.plsRcox.html">cv.plsRcox</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">X_train_micro</span>,time<span class="op">=</span><span class="va">Y_train_micro</span>,status<span class="op">=</span><span class="va">C_train_micro</span><span class="op">)</span>,nt<span class="op">=</span><span class="fl">10</span>,verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-cvplsRcox-1.png" title="plot of chunk cvplsRcox" alt="plot of chunk cvplsRcox" width="100%"></p>
<pre><code><span><span class="co">#&gt; $nt</span></span>
<span><span class="co">#&gt; [1] 10</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $cv.error5</span></span>
<span><span class="co">#&gt;  [1] 0.5000000 0.9674493 0.8840340 0.8881565 0.9611293 0.9694122 0.7785264</span></span>
<span><span class="co">#&gt;  [8] 0.7794468 0.7833874 0.7917907 0.7917344</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $cv.se5</span></span>
<span><span class="co">#&gt;  [1] 0.0000000000 0.0004328242 0.0371488864 0.0389160733 0.0007452107</span></span>
<span><span class="co">#&gt;  [6] 0.0040593349 0.0814540651 0.0815717378 0.0820171451 0.0829659086</span></span>
<span><span class="co">#&gt; [11] 0.0829564223</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds</span></span>
<span><span class="co">#&gt; $folds$`1`</span></span>
<span><span class="co">#&gt;  [1] 60  3  2 14 77  6 50  4 72 32 22  1 41 21 63 25</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds$`2`</span></span>
<span><span class="co">#&gt;  [1] 42 67 65 15 73 48 57 26  7 13 31 53  5 27 37 64</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds$`3`</span></span>
<span><span class="co">#&gt;  [1] 71 23 56 35 75 29 30 18 62 44 12 33 68 49 43 55</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds$`4`</span></span>
<span><span class="co">#&gt;  [1] 54 76 24 16 34 66  9 11 69 40 70 36 39  8 19 20</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $folds$`5`</span></span>
<span><span class="co">#&gt;  [1] 74 38 46 80 47 78 10 45 51 28 61 79 58 17 52 59</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $lambda.min5</span></span>
<span><span class="co">#&gt; [1] 5</span></span>
<span><span class="co">#&gt;</span></span>
<span><span class="co">#&gt; $lambda.1se5</span></span>
<span><span class="co">#&gt; [1] 0</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="dkplsrcox">DKplsRcox<a class="anchor" aria-label="anchor" href="#dkplsrcox"></a>
</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/DKplsRcox.html">DKplsRcox</a></span><span class="op">(</span><span class="va">X_train_micro</span>,time<span class="op">=</span><span class="va">Y_train_micro</span>,event<span class="op">=</span><span class="va">C_train_micro</span>,nt<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; Kernel :  rbfdot </span></span>
<span><span class="co">#&gt; Estimated_sigma  0.0122308</span></span>
<span><span class="co">#&gt; Error in model.matrix(mt0, mf0, contrasts.arg = contrasts.arg): l'argument "contrasts.arg" est manquant, avec aucune valeur par défaut</span></span>
<span></span>
<span><span class="fu"><a href="reference/DKplsRcox.html">DKplsRcox</a></span><span class="op">(</span><span class="op">~</span><span class="va">X_train_micro</span>,time<span class="op">=</span><span class="va">Y_train_micro</span>,event<span class="op">=</span><span class="va">C_train_micro</span>,nt<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in model.matrix(mt0, mf0, , contrasts.arg = contrasts.arg): l'argument "contrasts.arg" est manquant, avec aucune valeur par défaut</span></span>
<span></span>
<span><span class="fu"><a href="reference/DKplsRcox.html">DKplsRcox</a></span><span class="op">(</span>Xplan<span class="op">=</span><span class="va">X_train_micro</span>,time<span class="op">=</span><span class="va">Y_train_micro</span>,event<span class="op">=</span><span class="va">C_train_micro</span>,nt<span class="op">=</span><span class="fl">5</span>,sparse<span class="op">=</span><span class="cn">TRUE</span>, alpha.pvals.expli<span class="op">=</span><span class="fl">.15</span><span class="op">)</span></span>
<span><span class="co">#&gt; Kernel :  rbfdot </span></span>
<span><span class="co">#&gt; Estimated_sigma  0.01203267</span></span>
<span><span class="co">#&gt; Error in model.matrix(mt0, mf0, contrasts.arg = contrasts.arg): l'argument "contrasts.arg" est manquant, avec aucune valeur par défaut</span></span>
<span></span>
<span><span class="fu"><a href="reference/DKplsRcox.html">DKplsRcox</a></span><span class="op">(</span>Xplan<span class="op">=</span><span class="op">~</span><span class="va">X_train_micro</span>,time<span class="op">=</span><span class="va">Y_train_micro</span>,event<span class="op">=</span><span class="va">C_train_micro</span>,nt<span class="op">=</span><span class="fl">5</span>,sparse<span class="op">=</span><span class="cn">TRUE</span>, alpha.pvals.expli<span class="op">=</span><span class="fl">.15</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in model.matrix(mt0, mf0, , contrasts.arg = contrasts.arg): l'argument "contrasts.arg" est manquant, avec aucune valeur par défaut</span></span></code></pre></div>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=plsRcox" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/fbertran/plsRcox/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/fbertran/plsRcox/issues/" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing plsRcox</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Frederic Bertrand <br><small class="roles"> Maintainer, author </small> <a href="https://orcid.org/0000-0002-0837-8281" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li>Myriam Maumy-Bertrand <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-4615-1512" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://doi.org/10.32614/CRAN.package.plsRcox" class="external-link"><img src="https://img.shields.io/badge/doi-10.32614/CRAN.package.plsRcox-blue.svg" alt="DOI"></a></li>
<li><a href="https://lifecycle.r-lib.org/articles/stages.html" class="external-link"><img src="https://img.shields.io/badge/lifecycle-stable-green.svg" alt="Lifecycle: stable"></a></li>
<li><a href="https://www.repostatus.org/#active" class="external-link"><img src="https://www.repostatus.org/badges/latest/active.svg" alt="Project Status: Active – The project has reached a stable, usable state and is being actively developed."></a></li>
<li><a href="https://github.com/fbertran/plsRcox/actions" class="external-link"><img src="https://github.com/fbertran/plsRcox/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://app.codecov.io/gh/fbertran/plsRcox?branch=master" class="external-link"><img src="https://codecov.io/gh/fbertran/plsRcox/branch/master/graph/badge.svg" alt="Codecov test coverage"></a></li>
<li><a href="https://cran.r-project.org/package=plsRcox" class="external-link"><img src="https://www.r-pkg.org/badges/version/plsRcox" alt="CRAN status"></a></li>
<li><a href="https://cran.r-project.org/package=plsRcox" class="external-link"><img src="https://cranlogs.r-pkg.org/badges/plsRcox" alt="CRAN RStudio mirror downloads"></a></li>
<li><a href="https://github.com/fbertran/plsRcox" class="external-link"><img src="https://img.shields.io/github/stars/fbertran/plsRcox?style=social" alt="GitHub Repo stars"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Frederic Bertrand, Myriam Maumy-Bertrand.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
